---
title: "MYCN-NBL Subtyping"
author: "Aditya Lahiri, Eric Wafula, Jo Lynne Rokita"
date: "10/13/2022"
output: html_notebook
---
# Objective:
To molecularly subtype neuroblastoma, ganglioneuroblastoma, and ganglioneuroma samples into MYCN amplified or MYCN non-amplified.


# Usage
This notebook is intended to be run via the command line from the top directory
of the repository as follows:
```
Rscript -e "rmarkdown::render('analyses/molecular-subtyping-NBL/00-Analysis.Rmd', clean = TRUE)"
```
# Directories
```{r setup directories}
# root directory
root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))

# data directory
dataDir <- file.path(root_dir, "data")

#results directory
results_dir <- "results"
# create if doesn't exist
if (!dir.exists(results_dir)) {
  dir.create(results_dir)
}


#plot directory
plot_dir <- "plots"
# create if doesn't exist
if (!dir.exists(plot_dir)) {
  dir.create(plot_dir)
}
```

# Load Libraries
```{r load libraries}
library(dplyr)
library(data.table)
library(tidyverse)
library(ggplot2)
```

# Load all the datafiles in the ticket
```{r load the data files: this will take time}
hist_df<-readr::read_tsv(file.path(dataDir, "histologies-base.tsv"),guess_max = 100000)
consensus_df<-readr::read_tsv(file.path(dataDir, "consensus_wgs_plus_cnvkit_wxs.tsv.gz"),guess_max = 100000)
cnv_cnvkit_df<-readr::read_tsv(file.path(dataDir, "cnv-cnvkit.seg.gz"),guess_max = 100000)
cnv_controlfreec_df<-readr::read_tsv(file.path(dataDir, "cnv-controlfreec.tsv.gz"),guess_max = 100000)
gene_expression_df<-as.data.frame(readRDS(paste(dataDir,"/","gene-expression-rsem-tpm-collapsed.rds",sep="")))

```
# Part 1&2: 
1. For each NBL sample, gather MYCN amplification status from consensus_wgs_plus_cnvkit_wxs.tsv.gz
2. Create alterations/output table for matched DNA/RNA biospecimens. Match ID can be adapted from the independent-samples module

Code logic and background: 
NBL is defined as neuroblastoma, ganglioneuroblastoma, and ganglioneuroma.
These diseases are mentioned in the histology-base.tsv file. consensus_wgs_plus_cnvkit_wxs.tsv.gz 
has information regarding copy number, status, and gene symbol (MYCN). The key idea in the following 
code chunks is to find the MYCN samples that belong to NBL. Specifically for part 2 we need to use gene-expression-rsem-tpm-collapsed.rds file. This file has Biospecimen_ID_RNA for MYCN samples, and consensus_wgs_plus_cnvkit_wxs.tsv.gz has Biospecimen_ID_DNA for MYCN samples. The histologies files 
contain both DNA and RNA Biospecimen_ID. 


```{r Data filtering}
# First filter the histology file based on diseases we are intereseted in neuroblastoma, ganglioneuroblastoma, and ganglioneuroma
hist_filtered_df <- hist_df %>% filter(pathology_diagnosis=="Neuroblastoma"|
                                       pathology_diagnosis=="Ganglioneuroblastoma"|
                                       pathology_diagnosis=="Ganglioneuroblastoma, nodular"|
                                       pathology_diagnosis =="Ganglioneuroblastoma, intermixed"|
                                       pathology_diagnosis ==
                                         "Ganglioneuroma, maturing subtype OR Ganglioneuroblastoma, well differentiated")

# The consensus_wgs_plus_cnvkit_wxs.tsv.gz file has all types of genes, we just need MYCN, so we filter the dataset as follows:
consensus_filtered_df <- consensus_df %>% filter(gene_symbol=="MYCN")


# Get all the biospecimen IDs belonging to NBL (not necessarily MYCN-NBL, these biospecimen IDs can be for any gene but belong to NBL)
all_nbl_biospecimen_ID<- hist_filtered_df %>% select(Kids_First_Biospecimen_ID)

# Now get biospecimen dna ID for MYCN (but not necessarily belonging to NBL)
MYCN_biospecimen_dna_ID <- consensus_filtered_df %>% select(biospecimen_id)

# Get biospecimen rna ID for MYCN (but not necessarily belonging to NBL)
MYCN_Index <- which(rownames(gene_expression_df)=="MYCN") # find which rows (contains gene names) corresponds to MYCN
TPM_MYCN_data <- gene_expression_df[MYCN_Index,] # Get the TPM data for for MYCN
MYCN_biospecimen_rna_ID <- colnames(TPM_MYCN_data) # Get the column names which have biospecimen IDs
MYCN_biospecimen_rna_ID<-as.data.frame(MYCN_biospecimen_rna_ID) # Converting to dataframe for ease of use for later
```


```{r Changing column names}
colnames(MYCN_biospecimen_dna_ID)<- "Kids_First_Biospecimen_ID" #change biospecimen_ID to Kids_First_Biospecimen_ID
colnames(MYCN_biospecimen_rna_ID)<- "Kids_First_Biospecimen_ID" #change MYCN_biospecimen_rna_ID to Kids_First_Biospecimen_ID
# Store all the biospecimen IDs for MYCN both DNA and RNA in the following variable
all_MYCN_biospecimen_ID <- rbind(MYCN_biospecimen_dna_ID,MYCN_biospecimen_rna_ID)
```

```{r Getting TPM, Copy Number and Status info}
# Find which MYCN Biospecimen IDs in all_MYCN_biospecimen_ID correspond to NBL as defined in hist_filtered_df. Thus the intersection of biosepcimen in all_MYCN_biospecimen_ID  and hist_filtered_df will have the biospecimens IDs belonging to MYCN-NBL

MYCN_DF<- inner_join(hist_filtered_df,all_MYCN_biospecimen_ID,by="Kids_First_Biospecimen_ID")

#Add the satus and copy Number to these samples. Adding the information if available, so we do a left join with respect to MYCN_DF as we don't want to discard any biospecimen of MYCN-NBL for which Status and copy number is not avaialable
MYCN_DF<- left_join(MYCN_DF,consensus_filtered_df, by= c("Kids_First_Biospecimen_ID" = "biospecimen_id")) # Some rows increased likely copies

#Add TPM in  a similar fashion as copy number and status 
rownames(TPM_MYCN_data)<-"MYCN_TPM" # Defining the column names of tpm values
TPM_MYCN_data<-t(TPM_MYCN_data) # coverting 1 X N matrix to N X 1 matrix (lenght wise)
TPM_MYCN_data <-as.data.frame(TPM_MYCN_data) # Convert to data framte 
TPM_MYCN_data <- tibble::rownames_to_column(TPM_MYCN_data, "Kids_First_Biospecimen_ID") # Make ID rownames into a column 
# Finally add the TPM data using left join
MYCN_DF<- left_join(MYCN_DF,TPM_MYCN_data, by="Kids_First_Biospecimen_ID") # rows remain the same
```

Code logic and background: 
The data frame MYCN_DF has DNA and RNA specimen for a given sample, 
which will cause duplication, so we will find which DNA samples and 
RNA samples are related to each other in MYCN_DF. This is done by creating
an extra column called Match_ID in in MYCN_DF. Match_ID is 
bascially: Kids_First_Participant_ID + Sample_ID + Cohort.  
Match_ID will be same for corresponging DNA/RNA Samples. Cohort info is 
added as Kids_First_Participant_ID + Sample_ID can be same accross cohort.
```{r Find DNA/RNA Biospecimen}
# Create the MATCH ID : adapated from independent_samples module [1] 

MYCN_DF <- MYCN_DF %>%mutate(match_id = paste(Kids_First_Participant_ID, sample_id,cohort,sep = "_"))
# If the Biospecimen IDs are of DNA then the RNA_Library must be empty and experimental strategy is WGS, WXS or Targeted Sequencing 
MYCN_dna_df<-MYCN_DF %>% filter(is.na(RNA_library),experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing") ) 
# If the Biospecimen IDs are of DNA then experimental strategy is RNA-Seq
MYCN_rna_df<-MYCN_DF %>% filter(experimental_strategy == "RNA-Seq") 

# Rename the columns
colnames(MYCN_dna_df)[which(names(MYCN_dna_df) == "Kids_First_Biospecimen_ID")] <- "Kids_First_Biospecimen_ID_DNA"
colnames(MYCN_rna_df)[which(names(MYCN_rna_df) == "Kids_First_Biospecimen_ID")] <- "Kids_First_Biospecimen_ID_RNA"
```

Now find the intersection of MYCN_dna_df and MYCN_rna_df, this will give you a table of DNA and RNA biospecimen IDs
```{r Further filter the data to create alteration table}
MYCN_DNA_RNA_DF <- inner_join(MYCN_dna_df,MYCN_rna_df, by="match_id") 

# Retain the information of RNA/DNA Biospecimen ID, gene symbol, cancer group, pathology free text diagnosis, copy number, TPM and status
MYCN_DNA_RNA_FILTERED_DF <- MYCN_DNA_RNA_DF %>% select(Kids_First_Biospecimen_ID_DNA,Kids_First_Biospecimen_ID_RNA,gene_symbol.x,gene_symbol.y,
                                              pathology_diagnosis.x,pathology_diagnosis.y,
                                              pathology_free_text_diagnosis.x,pathology_free_text_diagnosis.y,
                                              copy_number.x,copy_number.y,MYCN_TPM.x,MYCN_TPM.y,status.x,status.y, match_id)
# Remove columns with no info
empty_columns <- colSums(is.na(MYCN_DNA_RNA_FILTERED_DF) | MYCN_DNA_RNA_FILTERED_DF == "") == nrow(MYCN_DNA_RNA_FILTERED_DF)
# Logic in empty column is adapted from [2] 
MYCN_DNA_RNA_FILTERED_DF<- MYCN_DNA_RNA_FILTERED_DF[!empty_columns]
#Remove duplicated coulmns
print(all.equal(MYCN_DNA_RNA_FILTERED_DF$pathology_diagnosis.x,MYCN_DNA_RNA_FILTERED_DF$pathology_diagnosis.y))
print(all.equal(MYCN_DNA_RNA_FILTERED_DF$pathology_free_text_diagnosis.x,MYCN_DNA_RNA_FILTERED_DF$pathology_free_text_diagnosis.y))

# The pathology free text diagnosis in row 114 of MYCN_DNA_RNA_FILTERED has differing values:
#print(which(mapply(identical,unlist(MYCN_DNA_RNA_FILTERED_DF$pathology_free_text_diagnosis.x),unlist(MYCN_DNA_RNA_FILTERED_DF$pathology_free_text_diagnosis.y))==FALSE))# Logic used from [3] 

#Since pathology_free_text_diagnosis.x is MYCN AMP and pathology_free_text_diagnosis.y is NA we will retain the information from pathology_free_text_diagnosis.x and remove pathology_free_text_diagnosis.y as it is identical to pathology_free_text_diagnosis.x otherwise. 

# Remove duplicate columns after inspection
duplicate_columns <- c("pathology_diagnosis.y","pathology_free_text_diagnosis.y")
MYCN_DNA_RNA_FILTERED_DF<-MYCN_DNA_RNA_FILTERED_DF[ , -which(names(MYCN_DNA_RNA_FILTERED_DF) %in% duplicate_columns)] # Reffered from the following code [4] 


# Clean the table:
colnames(MYCN_DNA_RNA_FILTERED_DF)<-c("Kids_First_Biospecimen_ID_DNA","Kids_First_Biospecimen_ID_RNA","gene_symbol",
                                      "cancer_group","pathology_free_text_diagnosis","copy_number","MYCN_TPM","status","Match_ID")    
```
Find the duplicate rows of MYCN_DNA_RNA_FILTERED_DF
```{r remove duplicate rows}
MYCN_DNA_RNA_FILTERED_DF<-distinct(MYCN_DNA_RNA_FILTERED_DF)
MYCN_Table <- as.data.frame(table(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA))
colnames(MYCN_Table)<- c("Kids_First_Biospecimen_ID_DNA", "Freq")
MYCN_Table$Kids_First_Biospecimen_ID_DNA<-as.character(MYCN_Table$Kids_First_Biospecimen_ID_DNA)
index_repeats <- which(MYCN_Table$Freq>1)
repeated_ids <- MYCN_Table$Kids_First_Biospecimen_ID_DNA[index_repeats]
for (iter in 1:length(repeated_ids))
  print(MYCN_DNA_RNA_FILTERED_DF[MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA==repeated_ids[iter],])
```
```{r removing values}
# For the record TARGET-30-PAPVFD-01A-01D retaining the record with status Neutral and copynumber 3
# For the record TARGET-30-PASXRG-01A-01D retaining the record with copy number 14
rows_to_delete_MYCN <- c(which(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA=="TARGET-30-PAPVFD-01A-01D" &
                               MYCN_DNA_RNA_FILTERED_DF$copy_number==2),which(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA
                                                                              =="TARGET-30-PASXRG-01A-01D" &
                                                                                MYCN_DNA_RNA_FILTERED_DF$copy_number==4))
MYCN_DNA_RNA_FILTERED_DF<- MYCN_DNA_RNA_FILTERED_DF[-rows_to_delete_MYCN,]
```



#Part 3
Add MYCN TPM and MYCN copy number and status to table. It may also be useful to see a quick ascending barplot of MYCN TPM (y-axis) and sample (x-axis) colored by clinical status (see 4 below). See example plot here.

```{r make plot}
#Select relevant info
MYCN_Plot_df <- MYCN_DNA_RNA_FILTERED_DF %>% select (Kids_First_Biospecimen_ID_DNA,MYCN_TPM,status)

#Binarize status column into amplification and non amplification

MYCN_Plot_df <- MYCN_Plot_df %>% mutate(status = str_replace(status, "neutral", "non-amplified"))
MYCN_Plot_df <- MYCN_Plot_df %>% mutate(status = str_replace(status, "gain", "non-amplified"))
MYCN_Plot_df <- MYCN_Plot_df %>% mutate(status = str_replace(status, "loss", "non-amplified"))

# Refered code from [5]
MYCN_Plot_df %>% 
  arrange(status,MYCN_TPM) %>% 
  mutate(Kids_First_Biospecimen_ID_DNA = factor(Kids_First_Biospecimen_ID_DNA, unique(Kids_First_Biospecimen_ID_DNA))) %>%
  ggplot() +aes(x=Kids_First_Biospecimen_ID_DNA, y=MYCN_TPM, fill=status)+
  geom_bar(position="dodge",stat="identity")+
  theme(axis.text.x = element_text(size = 15, angle = 45, hjust = 1))


plot_file <- file.path(plot_dir, "TPM_Biospecimen_Amp_NonAmp.png")
ggsave(filename = plot_file,width = 45, height = 20)



```

#Part 4: Assign subtype
```{r : select only relevant data}
alteration_df <- MYCN_DNA_RNA_FILTERED_DF %>% select(Kids_First_Biospecimen_ID_DNA,Kids_First_Biospecimen_ID_RNA,MYCN_TPM,
                                          copy_number,MYCN_TPM,status,pathology_free_text_diagnosis)
alteration_df$subtype<-NA

```

```{r : subtype assignment Case 1}

# ompare the clinical status (currently in pathology_free_text_diagnosis to the CNV call). If a match, assign molecular_subtype as NBL, MYCN amplified. If not a match: If MYCN is called as amplified but clinical file says non-amplified, assign NBL, MYCN amplified.

alteration_df<-alteration_df %>%mutate(subtype = case_when(status =="amplification" & pathology_free_text_diagnosis=="MYCN amp" 
                                                           ~ "NBL, MYCN amplified",status =="amplification" &
                                                             pathology_free_text_diagnosis %in% 
                                                             c("Metastatic secondary tumors;Neuroblastoma","MYCN non-amp",
                                                               NA_character_) ~ "NBL, MYCN amplified",
                                       status %in% c("gain", "loss", "neutral") & 
                                         pathology_free_text_diagnosis %in% c("Metastatic secondary tumors;Neuroblastoma", 
                                                                              "MYCN non-amp", NA_character_)~ 
                                         "NBL, MYCN non amplified"))
                                            
```

```{r subtype assignment Case}
# If MYCN is called non-amplified but clinical file says amplified, plot the CNV data to visualize whether we see focal amplification but the CNV was not called. 

index_clinical_amplified <- which(alteration_df$status %in% c("gain","loss")  
                                  & alteration_df$pathology_free_text_diagnosis=="MYCN amp")

clincal_amplified_IDs <- alteration_df$Kids_First_Biospecimen_ID_DNA[index_clinical_amplified]

```

```{r plot function loading}

# adapating code from: https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/785c3224de29f701b1c1b62841d0f84d46f7eaca/analyses/molecular-subtyping-embryonal/03-clean-c19mc-data.Rmd#L56-L112


plot_chr2 <- function(cn_df, biospecimen_id) {
 # This function takes a seg data.frame and a biospecimen identifier and 
 # returns a plot of the chromosome 19 segment mean for that biospecimen.
 # It fills missing seg.mean values with zeroes.
 # 
 # Args:
 #   cn_df: data.frame in SEG format
 #   biospecimen_id: A Kids First Biospecimen ID used to filter cn_df
 # 
 # Returns: essentially a barplot of the non-neutral segment means
  
  bsid_data <- cn_df %>%  
    # For visualization purposes fill the missing seg.mean values with zeroes
    tidyr::replace_na(list(seg.mean = 0)) %>%
    # Reformat the chromosome variable to drop the "chr"
    dplyr::mutate(chrom = factor(gsub("chr", "", chrom), 
                                 levels = c(1:22, "X", "Y"))) %>%
    # Only look at chr19 in the relevant sample
    filter(ID == biospecimen_id,
           chrom == 2) %>%
    # Make Del/Amp variable
    dplyr::mutate(Type = dplyr::case_when(
      seg.mean < 0 ~ "Del",
      seg.mean > 0 ~ "Amp",
      seg.mean == 0 ~ "Neutral"
    ))
  
  # Turn into a GRanges for easier mapping
  bsid_ranges <- GenomicRanges::GRanges(
    seqnames = bsid_data$chrom,
    ranges = IRanges::IRanges(
      start = bsid_data$loc.start,
      end = bsid_data$loc.end
    ),
    score = bsid_data$seg.mean,
    mcols = bsid_data$Type
  )
  
  # Map this on a plot
  bsid_plot <- 
    ggbio::autoplot(bsid_ranges,
                    ggplot2::aes(y = score, fill = mcols),
                    geom = "bar") +
    ggplot2::theme_bw() +
    ggplot2::ylim(c(-2, 2)) +
    colorblindr::scale_fill_OkabeIto(name = "Type") +
    ggplot2::labs(
      title = paste(biospecimen_id, "chr2"),
      y = "segment mean"
    )
  
  return(bsid_plot)
}

```


```{r plots}
plot_chr2(cnv_cnvkit_df,clincal_amplified_IDs[1])
```
```{r}
plot_chr2(cnv_cnvkit_df,clincal_amplified_IDs[2])
```



```{r}
get_status <- function (cn_df, biospecimen_id){
  status <- "NBL, MYCN non amplified"
  cn_df<- cn_df[cn_df$ID==biospecimen_id,]
  mean_segmean<- mean(cn_df$seg.mean)
  if(mean_segmean>0){
    status<-"NBL, MYCN amplified"
  }
 return(status)
}

```

```{r}
for (iter in 1:length(clincal_amplified_IDs)){
  alteration_df$subtype[index_clinical_amplified[iter]]<- get_status(cnv_cnvkit_df,clincal_amplified_IDs[iter])
}

```



Cases when status call is neutral and pathology_free_text_diagnosis is amplified.

```{r}
index_clinical_neutral <- which(alteration_df$status =="neutral"  
                                  & alteration_df$pathology_free_text_diagnosis=="MYCN amp")
clincal_neutral_IDs <- alteration_df$Kids_First_Biospecimen_ID_DNA[index_clinical_neutral]

```


#Part 5: Write the table
```{r write alteration table}
alteration_df%>%readr::write_tsv(file.path(results_dir, "Alteration_Table.tsv"))
```

# References
[1]https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/independent-samples/util/independent-rna-samples.R

[2]https://www.codingprof.com/3-easy-ways-to-remove-empty-columns-in-r/#:~:text=Identify%20the%20empty%20columns,empty%20values%20in%20a%20column.

[3] https://stat.ethz.ch/pipermail/r-help/2010-April/234706.html

[4]https://stackoverflow.com/questions/5234117/how-to-drop-columns-by-name-in-a-data-frame

[5]https://stackoverflow.com/questions/64638958/grouped-bars-in-descending-order