---
title: "MYCN-NBL Subtyping"
author: "Aditya Lahiri, Eric Wafula, Jo Lynne Rokita"
date: "10/13/2022"
output: html_notebook
---
# Objective:
To molecularly subtype neuroblastoma, ganglioneuroblastoma, and ganglioneuroma samples into MYCN amplified or MYCN non-amplified.


# Usage
This notebook is intended to be run via the command line from the top directory
of the repository as follows:
```
Rscript -e "rmarkdown::render('analyses/molecular-subtyping-NBL/00-Analysis.Rmd', clean = TRUE)"
```
# Directories
```{r setup directories}
# root directory
root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))

# data directory
dataDir <- file.path(root_dir, "data")

# module directory
moduleDir <- file.path(root_dir, "analyses","molecular-subtyping-NBL")

#results directory
results_dir <- "results"
# create if doesn't exist
if (!dir.exists(results_dir)) {
  dir.create(results_dir)
}


#plot directory
plot_dir <- "plots"
# create if doesn't exist
if (!dir.exists(plot_dir)) {
  dir.create(plot_dir)
}
```

# Load Libraries
```{r load libraries}
library(dplyr)
library(data.table)
library(tidyverse)
library(ggplot2)
```

# Load all the datafiles in the ticket
```{r load the data files: this will take time}
hist_df<-readr::read_tsv(file.path(dataDir, "histologies-base.tsv"),guess_max = 100000)
consensus_df<-readr::read_tsv(file.path(dataDir, "consensus_wgs_plus_cnvkit_wxs.tsv.gz"),guess_max = 100000)
cnv_cnvkit_df<-readr::read_tsv(file.path(dataDir, "cnv-cnvkit.seg.gz"),guess_max = 100000)
cnv_controlfreec_df<-readr::read_tsv(file.path(dataDir, "cnv-controlfreec.tsv.gz"),guess_max = 100000)
gene_expression_df<-as.data.frame(readRDS(paste(dataDir,"/","gene-expression-rsem-tpm-collapsed.rds",sep="")))
gmkf_patient_clinical_mycn_status_df <- readr::read_tsv(file.path(moduleDir, "input/gmkf_patient_clinical_mycn_status.tsv"),guess_max = 100000)
target_patient_clinical_mycn_status_df<- readr::read_tsv(file.path(moduleDir, "input/target_patient_clinical_mycn_status.tsv"),guess_max = 100000)
```
# Part 1&2: 
1. For each NBL sample, gather MYCN amplification status from consensus_wgs_plus_cnvkit_wxs.tsv.gz
2. Create alterations/output table for matched DNA/RNA biospecimens. Match ID can be adapted from the independent-samples module

Code logic and background: 
NBL is defined as neuroblastoma, ganglioneuroblastoma, and ganglioneuroma.These diseases are mentioned in the histology-base.tsv file. 

consensus_wgs_plus_cnvkit_wxs.tsv.gz has information regarding copy number, status, and gene symbol (MYCN). 

The key idea in the following code chunks is to find the MYCN samples that belong to NBL. Specifically for part 2 we need to use gene-expression-rsem-tpm-collapsed.rds file. This file has Biospecimen_ID_RNA for MYCN samples, and consensus_wgs_plus_cnvkit_wxs.tsv.gz has Biospecimen_ID_DNA for MYCN samples. The histologies files 
contain both DNA and RNA Biospecimen_ID. 


```{r Data filtering}
# First filter the histology file based on diseases we are intereseted in neuroblastoma, ganglioneuroblastoma, and ganglioneuroma
hist_filtered_df <- hist_df %>% filter(pathology_diagnosis=="Neuroblastoma"|
                                       pathology_diagnosis=="Ganglioneuroblastoma"|
                                       pathology_diagnosis=="Ganglioneuroblastoma, nodular"|
                                       pathology_diagnosis =="Ganglioneuroblastoma, intermixed"|
                                       pathology_diagnosis ==
                                         "Ganglioneuroma, maturing subtype OR Ganglioneuroblastoma, well differentiated",
                                       sample_type=="Tumor",experimental_strategy %in%
                                         c("WGS","WXS","Targeted Sequencing","RNA-Seq"))

# The consensus_wgs_plus_cnvkit_wxs.tsv.gz file has all types of genes, we just need MYCN, so we filter the dataset as follows:
consensus_filtered_df <- consensus_df %>% filter(gene_symbol=="MYCN")


# Now get biospecimen dna ID for MYCN (but not necessarily belonging to NBL)
MYCN_Consensus_ID <- consensus_filtered_df %>% select(biospecimen_id)

# Get biospecimen rna ID for MYCN (but not necessarily belonging to NBL)
MYCN_Index <- which(rownames(gene_expression_df)=="MYCN") # find which rows (contains gene names) corresponds to MYCN
TPM_MYCN_data <- gene_expression_df[MYCN_Index,] # Get the TPM data for for MYCN
MYCN_TPM_ID <- colnames(TPM_MYCN_data) # Get the column names which have biospecimen IDs
MYCN_TPM_ID<-as.data.frame(MYCN_TPM_ID) # Converting to dataframe for ease of use for later

colnames(MYCN_Consensus_ID)<- "Kids_First_Biospecimen_ID" #change biospecimen_ID to Kids_First_Biospecimen_ID
colnames(MYCN_TPM_ID)<- "Kids_First_Biospecimen_ID" #change MYCN_TPM_ID to Kids_First_Biospecimen_ID
# Store all the biospecimen IDs for MYCN both DNA and RNA in the following variable
all_MYCN_biospecimen_ID <- rbind(MYCN_Consensus_ID,MYCN_TPM_ID)
```


```{r Getting TPM, Copy Number and Status info}
# Find which MYCN Biospecimen IDs in all_MYCN_biospecimen_ID correspond to NBL as defined in hist_filtered_df. 
MYCN_DF<- left_join(hist_filtered_df,all_MYCN_biospecimen_ID,by="Kids_First_Biospecimen_ID") ##### 1508 samples 

#Add the satus and copy Number to these samples. Adding the information if available, so we do a left join with respect to MYCN_DF as we don't want to discard any biospecimen of MYCN-NBL for which Status and copy number is not avaialable
MYCN_DF<- left_join(MYCN_DF,consensus_filtered_df, by= c("Kids_First_Biospecimen_ID" = "biospecimen_id")) # Some rows increased likely copies , 1514 samples , there were 1508 samples

#Add TPM in  a similar fashion as copy number and status 
rownames(TPM_MYCN_data)<-"MYCN_TPM" # Defining the column names of tpm values
TPM_MYCN_data<-t(TPM_MYCN_data) # coverting 1 X N matrix to N X 1 matrix (lenght wise)
TPM_MYCN_data <-as.data.frame(TPM_MYCN_data) # Convert to data framte 
TPM_MYCN_data <- tibble::rownames_to_column(TPM_MYCN_data, "Kids_First_Biospecimen_ID") # Make ID rownames into a column 
# Finally add the TPM data using left join
MYCN_DF<- left_join(MYCN_DF,TPM_MYCN_data, by="Kids_First_Biospecimen_ID") 

MYCN_DF<- distinct(MYCN_DF) # 1508 samples 
```

Code logic and background: 
The data frame MYCN_DF has DNA and RNA specimen for a given sample, 
which will cause duplication, so we will find which DNA samples and 
RNA samples are related to each other in MYCN_DF. This is done by creating
an extra column called Match_ID in in MYCN_DF. Match_ID is 
bascially: Kids_First_Participant_ID + Sample_ID + Cohort.  
Match_ID will be same for corresponging DNA/RNA Samples. Cohort info is 
added as Kids_First_Participant_ID + Sample_ID can be same accross cohort.
```{r Find DNA/RNA Biospecimen}
# Create the MATCH ID : adapated from independent_samples module [1] 

MYCN_DF <- MYCN_DF %>%mutate(match_id = paste(Kids_First_Participant_ID, sample_id,cohort,sep = "_"))
# If the Biospecimen IDs are of DNA then the RNA_Library must be empty and experimental strategy is WGS, WXS or Targeted Sequencing 
#MYCN_dna_df<-MYCN_DF %>% filter(is.na(RNA_library),experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing"))
# MYCN_dna has 1103 samples
MYCN_dna_df<-MYCN_DF %>% filter(is.na(RNA_library),experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing")) # 1103 samples samples

# If the Biospecimen IDs are of DNA then experimental strategy is RNA-Seq
#MYCN_rna_df<-MYCN_DF %>% filter(experimental_strategy == "RNA-Seq", (!is.na(RNA_library) & experimental_strategy=="Targeted Sequencing")) # 374 samples
MYCN_rna_df<-MYCN_DF %>% filter( (experimental_strategy=="RNA-Seq") | (experimental_strategy=="Targeted Sequencing" & !is.na(RNA_library))) # 374 rna-seq + 31 targeted_seq+exome_capture

# Rename the columns
colnames(MYCN_dna_df)[which(names(MYCN_dna_df) == "Kids_First_Biospecimen_ID")] <- "Kids_First_Biospecimen_ID_DNA"
colnames(MYCN_rna_df)[which(names(MYCN_rna_df) == "Kids_First_Biospecimen_ID")] <- "Kids_First_Biospecimen_ID_RNA"
```

Now find the intersection of MYCN_dna_df and MYCN_rna_df, this will give you a table of DNA and RNA biospecimen IDs
```{r Further filter the data to create alteration table}
MYCN_DNA_RNA_DF <- inner_join(MYCN_dna_df,MYCN_rna_df, by="match_id")  # 312 samples with DNA and RNA IDs 342 samples after exome cap

# Retain the information of RNA/DNA Biospecimen ID, gene symbol, cancer group, pathology free text diagnosis, copy number, TPM and status
MYCN_DNA_RNA_FILTERED_DF <- MYCN_DNA_RNA_DF %>% select(Kids_First_Biospecimen_ID_DNA,Kids_First_Biospecimen_ID_RNA,Kids_First_Participant_ID.x,Kids_First_Participant_ID.y,
                                              gene_symbol.x,gene_symbol.y,
                                              pathology_diagnosis.x,pathology_diagnosis.y,
                                              pathology_free_text_diagnosis.x,pathology_free_text_diagnosis.y,
                                              copy_number.x,copy_number.y,MYCN_TPM.x,MYCN_TPM.y,status.x,status.y, match_id)
# Remove columns with no info
empty_columns <- colSums(is.na(MYCN_DNA_RNA_FILTERED_DF) | MYCN_DNA_RNA_FILTERED_DF == "") == nrow(MYCN_DNA_RNA_FILTERED_DF)
# Logic in empty column is adapted from [2] 
MYCN_DNA_RNA_FILTERED_DF<- MYCN_DNA_RNA_FILTERED_DF[!empty_columns]
#Remove duplicated coulmns
print(all.equal(MYCN_DNA_RNA_FILTERED_DF$pathology_diagnosis.x,MYCN_DNA_RNA_FILTERED_DF$pathology_diagnosis.y))
print(all.equal(MYCN_DNA_RNA_FILTERED_DF$pathology_free_text_diagnosis.x,MYCN_DNA_RNA_FILTERED_DF$pathology_free_text_diagnosis.y))
print(all.equal(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Participant_ID.x,MYCN_DNA_RNA_FILTERED_DF$Kids_First_Participant_ID.y))


# Remove duplicate columns after inspection
duplicate_columns <- c("pathology_diagnosis.y","pathology_free_text_diagnosis.y","Kids_First_Participant_ID.y")
MYCN_DNA_RNA_FILTERED_DF<-MYCN_DNA_RNA_FILTERED_DF[ , -which(names(MYCN_DNA_RNA_FILTERED_DF) %in% duplicate_columns)] # Reffered from the following code [4]  312 samples are present. 342 samples 


# Rename the columns appropriately.
colnames(MYCN_DNA_RNA_FILTERED_DF)<-c("Kids_First_Biospecimen_ID_DNA","Kids_First_Biospecimen_ID_RNA",
                                      "Kids_First_Participant_ID","gene_symbol","cancer_group",
                                      "pathology_free_text_diagnosis","copy_number","MYCN_TPM","status","Match_ID")    
```
Find the repeating Kids_First_Biospecimen_ID_DNA MYCN_DNA_RNA_FILTERED_DF
```{r check for repeating Kids_First_Biospecimen_ID_DNA}
# We will check only repeating DNA IDs as the records unique in MYCN_DNA_RNA_FILTERED_DF when we query them with 
# DNA IDs + RNA IDs. 
MYCN_Table <- as.data.frame(table(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA))
colnames(MYCN_Table)<- c("Kids_First_Biospecimen_ID_DNA", "Freq")
MYCN_Table$Kids_First_Biospecimen_ID_DNA<-as.character(MYCN_Table$Kids_First_Biospecimen_ID_DNA)
index_repeats <- which(MYCN_Table$Freq>1)
repeated_ids <- MYCN_Table$Kids_First_Biospecimen_ID_DNA[index_repeats]
for (iter in 1:length(repeated_ids))
  print(MYCN_DNA_RNA_FILTERED_DF[MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA==repeated_ids[iter],])
```
```{r removing values}
# For the record TARGET-30-PAPVFD-01A-01D retaining the record with status Neutral and copynumber 3
# For the record TARGET-30-PASXRG-01A-01D retaining the record with copy number 14

# For records BS_0XC02E11 and BS_1F8J25Q1 two samples are there, differing in alliquot.id.y
rows_to_delete_MYCN <- c(which(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA=="TARGET-30-PAPVFD-01A-01D" 
                        & MYCN_DNA_RNA_FILTERED_DF$copy_number==2),
                        which(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA=="TARGET-30-PASXRG-01A-01D" &
                                                                              MYCN_DNA_RNA_FILTERED_DF$copy_number==4))
rows_to_delete_MYCN_DGD_1<-which(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA=="BS_0XC02E11") [2] # remove ET_FD9T78QE_DGD_FUSIP_29
rows_to_delete_MYCN_DGD_2<-which(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA=="BS_1F8J25Q1") [2] # ET_FD9T78QE_DGD_FUSIP_29 
MYCN_DNA_RNA_FILTERED_DF<- MYCN_DNA_RNA_FILTERED_DF[-c(rows_to_delete_MYCN,rows_to_delete_MYCN_DGD_1,rows_to_delete_MYCN_DGD_2),] # 338 samples 
```



Plot TPMs vs Biospecimen for samples that have both DNA and RNA IDs

```{r make TPM vs Biospecimen plot for samples with DNA and RNA IDs}
#Select relevant info
MYCN_Plot_df <- MYCN_DNA_RNA_FILTERED_DF %>% select (Kids_First_Biospecimen_ID_DNA,MYCN_TPM,status)

# Refered code from [5]
bar_plot1<-MYCN_Plot_df %>% 
  arrange(MYCN_TPM) %>% 
  mutate(Kids_First_Biospecimen_ID_DNA = factor(Kids_First_Biospecimen_ID_DNA, unique(Kids_First_Biospecimen_ID_DNA))) %>%
  ggplot() +aes(x=Kids_First_Biospecimen_ID_DNA, y=MYCN_TPM, fill=status)+
  geom_bar(position="dodge",stat="identity")+
  theme(axis.text.x = element_text(size = 5, angle = 45, hjust = 1))

#bar_plot1+ geom_hline(aes(yintercept=Suggested_Cutoff,linetype = "Suggested Amp Cutoff"),color = "black",size=2) #TARGET-30-PAPKXS-01A-01D	

plot_file <- file.path(plot_dir, "TPM_Biospecimen_Matching.png")
ggsave(filename = plot_file,width = 45, height = 20)



```


# add IDs that don't necessarily have both DNA and RNA 

```{r Gather samples that dont have both DNA and RNA IDs}
MYCN_IDs_exhaustive <- as.data.frame(unique(MYCN_DF$Kids_First_Biospecimen_ID)) # Has every biospecimen ID of NBL-MYCN samples
colnames(MYCN_IDs_exhaustive)<-"Kids_First_Biospecimen_ID"
# Repeating IDs with different copy numbers TARGET-30-PAMDAL-01A-01W, TARGET-30-PAPVFD-01A-01D, TARGET-30-PASXRG-01A-01D  totally 1508 samples

# Last two are sorted out in DNA-RNA matching 


MYCN_IDs_with_DNA_RNA <- as.data.frame(c(MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_DNA,
                               MYCN_DNA_RNA_FILTERED_DF$Kids_First_Biospecimen_ID_RNA)) # 624 samples but half of them are DNA and rest are corresponding RNA  
colnames(MYCN_IDs_with_DNA_RNA)<-"Kids_First_Biospecimen_ID"

MYCN_IDs_without_DNA_RNA_Match <- as.data.frame(MYCN_IDs_exhaustive[-(which(MYCN_IDs_exhaustive$Kids_First_Biospecimen_ID %in% MYCN_IDs_with_DNA_RNA$Kids_First_Biospecimen_ID)),]) # 886 samples here 
colnames(MYCN_IDs_without_DNA_RNA_Match)<- "Kids_First_Biospecimen_ID"
# 1508-624= 884 but remember TARGET-30-PAPVFD-01A-01D, TARGET-30-PASXRG-01A-01D has two copies in MYCN_IDs_exhaustive so extra 2 samples are further removed so 378 samples without RNA


# Now we need to identify which IDs in MYCN_IDs_without_DNA_RNA_Match are DNA IDs and Which are RNA IDs
DNA_IDs_MYCN_Non_Match <- as.data.frame(MYCN_IDs_without_DNA_RNA_Match[which(MYCN_IDs_without_DNA_RNA_Match$Kids_First_Biospecimen_ID %in% MYCN_dna_df$Kids_First_Biospecimen_ID_DNA),])
colnames(DNA_IDs_MYCN_Non_Match)<- "Kids_First_Biospecimen_ID_DNA"

RNA_IDs_MYCN_Non_Match <- as.data.frame(MYCN_IDs_without_DNA_RNA_Match[which(MYCN_IDs_without_DNA_RNA_Match$Kids_First_Biospecimen_ID %in% MYCN_rna_df$Kids_First_Biospecimen_ID_RNA),])
colnames(RNA_IDs_MYCN_Non_Match)<- "Kids_First_Biospecimen_ID_RNA"

# Make dfs DNA and RNA IDs found here with  

DNA_only_MYCN_df<- MYCN_DF %>% filter(Kids_First_Biospecimen_ID %in% DNA_IDs_MYCN_Non_Match$Kids_First_Biospecimen_ID_DNA)

DNA_only_MYCN_df <- DNA_only_MYCN_df %>% select(Kids_First_Biospecimen_ID,Kids_First_Participant_ID,MYCN_TPM,copy_number,
                                                status,pathology_free_text_diagnosis)
colnames(DNA_only_MYCN_df)[1]<- "Kids_First_Biospecimen_ID_DNA"


RNA_only_MYCN_df<- MYCN_DF %>% filter(Kids_First_Biospecimen_ID %in% RNA_IDs_MYCN_Non_Match$Kids_First_Biospecimen_ID_RNA)

RNA_only_MYCN_df <- RNA_only_MYCN_df %>% select(Kids_First_Biospecimen_ID,Kids_First_Participant_ID,MYCN_TPM,
                                                copy_number,status,pathology_free_text_diagnosis)
colnames(RNA_only_MYCN_df)[1]<- "Kids_First_Biospecimen_ID_RNA"


# Add biospecimen_dna_id and biospecimen_rna_id accordingly

DNA_only_MYCN_df$Kids_First_Biospecimen_ID_RNA<- NA_character_
RNA_only_MYCN_df$Kids_First_Biospecimen_ID_DNA<- NA_character_
DNA_only_MYCN_df<- DNA_only_MYCN_df[,c(1,7,2:6)] # Rearrange the columns to make it look like MYCN_DNA_RNA_FILTERED_DF
RNA_only_MYCN_df<- RNA_only_MYCN_df[,c(7,1:6)] 

# Combine the above two dfs, these are DFs where either DNA or RNA Biospecimen_ID is missing
MYCN_non_match_df <- rbind(DNA_only_MYCN_df,RNA_only_MYCN_df)
```


#Part 4: Assign subtype
```{r : select only relevant data for subtyping table}
alteration_df <- MYCN_DNA_RNA_FILTERED_DF %>% select(Kids_First_Biospecimen_ID_DNA,Kids_First_Biospecimen_ID_RNA,                                              Kids_First_Participant_ID,MYCN_TPM,copy_number,MYCN_TPM,status,pathology_free_text_diagnosis)
#alteration_df <- alteration_df %>% mutate(pathology_free_text_diagnosis2 = pathology_free_text_diagnosis)


alteration_df <- rbind(alteration_df,MYCN_non_match_df)


# Bring in the extra pathology free diagnonsis text from files in input folder:
colnames(gmkf_patient_clinical_mycn_status_df)<-c("Kids_First_Participant_ID","pathology_free_text_diagnosis_gmfk")
colnames(target_patient_clinical_mycn_status_df)<-c("Kids_First_Participant_ID","pathology_free_text_diagnosis_target")

alteration_df<- left_join(alteration_df,gmkf_patient_clinical_mycn_status_df,by="Kids_First_Participant_ID")
alteration_df<- left_join(alteration_df,target_patient_clinical_mycn_status_df,by="Kids_First_Participant_ID")
alteration_df<-alteration_df%>% mutate(pathology=coalesce(pathology_free_text_diagnosis,pathology_free_text_diagnosis_gmfk,
                                                          pathology_free_text_diagnosis_target))
alteration_df<-alteration_df %>% select(Kids_First_Biospecimen_ID_DNA,Kids_First_Biospecimen_ID_RNA,                            
                                        MYCN_TPM,copy_number,MYCN_TPM,status,pathology)
colnames(alteration_df)[6]<- "pathology_free_text_diagnosis"
alteration_df$subtype<-NA # Create a subtype column 

```

```{r Plot of all biospecimen with TPMs}
plot_alteration_df <- alteration_df %>% filter(!is.na(MYCN_TPM))
plot_alteration_df<- plot_alteration_df %>% mutate(Kids_First_Biospecimen_ID=coalesce(Kids_First_Biospecimen_ID_DNA,Kids_First_Biospecimen_ID_RNA))

bar_plot2<-plot_alteration_df %>% 
  arrange(MYCN_TPM) %>% 
  mutate(Kids_First_Biospecimen_ID = factor(Kids_First_Biospecimen_ID, unique(Kids_First_Biospecimen_ID))) %>%
  ggplot() +aes(x=Kids_First_Biospecimen_ID, y=MYCN_TPM, fill=status)+
  geom_bar(position="dodge",stat="identity")+
  theme(axis.text.x = element_text(size = 2, angle = 45, hjust = 1))
bar_plot2

Suggested_Cutoff<-140.83 # TARGET-30-PAMEZH-01A-01R	
bar_plot2+ geom_hline(aes(yintercept=Suggested_Cutoff,linetype = "Suggested Amp Cutoff"),color = "black",size=2) #TARGET-30-PAPKXS-01A-01D	

plot_file <- file.path(plot_dir, "TPM_Biospecimen_All_Samples_With_TMP.png")
ggsave(filename = plot_file,width = 49, height = 20)


```






















```{r Subtyping: when match is found and when call is amp and pathology free text is non amp}
alteration_df<-alteration_df %>%mutate(subtype = case_when(status =="amplification" & pathology_free_text_diagnosis=="MYCN amp"
                                                  ~ "NBL, MYCN amplified",status =="amplification" & 
                                                    pathology_free_text_diagnosis %in% c("MYCN non-amp","ganglioneuroblastoma, stage ii favorable histology non n-myc amplified") ~ "NBL, MYCN amplified",
                                       status %in% c("gain","loss","neutral") &  pathology_free_text_diagnosis %in% c("MYCN non-amp","ganglioneuroblastoma, stage ii favorable histology non n-myc amplified") ~ "NBL, MYCN non-amplified"))



```



```{r subtyping when call is not amp but pathology free text is amp}
# If MYCN is called non-amplified but clinical file says amplified, plot the CNV data to visualize whether we see focal amplification but the CNV was not called. 

index_clinical_amplified <- which(alteration_df$status %in% c("gain","loss","neutral")  
                                  & alteration_df$pathology_free_text_diagnosis=="MYCN amp")

clincal_amplified_IDs <- alteration_df$Kids_First_Biospecimen_ID_DNA[index_clinical_amplified]


print("Biospecimens for which pathology (clinical) file says  amplified but call is loss, gain or netural")
print(clincal_amplified_IDs)
```


```{r plot function loading}

# adapating code from: https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/785c3224de29f701b1c1b62841d0f84d46f7eaca/analyses/molecular-subtyping-embryonal/03-clean-c19mc-data.Rmd#L56-L112

# Apply this code to chromosome 2 since MYCN is on Chromosome 2

plot_chr2 <- function(cn_df, biospecimen_id) {
 # This function takes a seg data.frame and a biospecimen identifier and 
 # returns a plot of the chromosome 2 segment mean for that biospecimen.
 # It fills missing seg.mean values with zeroes.
 # 
 # Args:
 #   cn_df: data.frame in SEG format
 #   biospecimen_id: A Kids First Biospecimen ID used to filter cn_df
 # 
 # Returns: essentially a barplot of the non-neutral segment means
  
  bsid_data <- cn_df %>%  
    # For visualization purposes fill the missing seg.mean values with zeroes
    tidyr::replace_na(list(seg.mean = 0)) %>%
    # Reformat the chromosome variable to drop the "chr"
    dplyr::mutate(chrom = factor(gsub("chr", "", chrom), 
                                 levels = c(1:22, "X", "Y"))) %>%
    # Only look at chr2 in the relevant sample
    filter(ID == biospecimen_id,
           chrom == 2) %>%
    # Make Del/Amp variable
    dplyr::mutate(Type = dplyr::case_when(
      seg.mean < 0 ~ "Del",
      seg.mean > 0 ~ "Amp",
      seg.mean == 0 ~ "Neutral"
    ))
  
  # Turn into a GRanges for easier mapping
  bsid_ranges <- GenomicRanges::GRanges(
    seqnames = bsid_data$chrom,
    ranges = IRanges::IRanges(
      start = bsid_data$loc.start,
      end = bsid_data$loc.end
    ),
    score = bsid_data$seg.mean,
    mcols = bsid_data$Type
  )
  
  # Map this on a plot
  bsid_plot <- 
    ggbio::autoplot(bsid_ranges,
                    ggplot2::aes(y = score, fill = mcols),
                    geom = "bar") +
    ggplot2::theme_bw() +
    ggplot2::ylim(c(-2, 2)) +
    colorblindr::scale_fill_OkabeIto(name = "Type") +
    ggplot2::labs(
      title = paste(biospecimen_id, "chr2"),
      y = "segment mean"
    )
  
  return(bsid_plot)
}

```


```{r create the segmean vs location plots for samples where call is non-amp but pathology free text is amp}
for (iter in 1: length(clincal_amplified_IDs)){
  if(!is.na(clincal_amplified_IDs[iter])){
    plot_chr2(cnv_cnvkit_df,clincal_amplified_IDs[iter])
    plt_file <- file.path(plot_dir, paste(clincal_amplified_IDs[iter],"_Amp_Del",".png",sep=""))
    ggsave(filename = plt_file,width = 10, height = 20)
  }
  else{
    plot_chr2(cnv_cnvkit_df,alteration_df$Kids_First_Biospecimen_ID_RNA[index_clinical_amplified[iter]])
    plt_file <- file.path(plot_dir, paste(clincal_amplified_IDs[iter],"_Amp_Del",".png",sep=""))
    ggsave(filename = plt_file,width = 10, height = 20)
  }
  
}
```


```{r : Assign Subtypes status non amp but pathology free text amp}
index_clincal <- NA
for (iter in index_clinical_amplified){
  if(is.na(alteration_df$MYCN_TPM[iter])){
    alteration_df$subtype[iter]<- "Pathology-amp,Status-non-amp,TPM-NA"

  }
  else if (alteration_df$MYCN_TPM[iter] >= Suggested_Cutoff){
   alteration_df$subtype[iter]<- "NBL, MYCN amplified"
   index_clincal <- c(index_clincal,iter)

 }
  else if (alteration_df$MYCN_TPM[iter] < Suggested_Cutoff) {
    alteration_df$subtype[iter]<- "NBL, MYCN non-amplified"
    index_clincal <- c(index_clincal,iter)

  } 
  
}
```




```{r : Assign the remaining unclassified samples based of a suggested cutoff TPM }
index_tpm_unsubtype_above<-which(is.na(alteration_df$subtype) & !is.na(alteration_df$MYCN_TPM)& alteration_df$MYCN_TPM >= Suggested_Cutoff)

index_tpm_unsubtype_below<-which(is.na(alteration_df$subtype) & !is.na(alteration_df$MYCN_TPM)& alteration_df$MYCN_TPM < Suggested_Cutoff)

alteration_df$subtype[index_tpm_unsubtype_above]<-"NBL, MYCN amplified"

alteration_df$subtype[index_tpm_unsubtype_below]<-"NBL, MYCN non-amplified"

```


```{r}
index_unclassified <- which(is.na(alteration_df$subtype))
alteration_df$subtype[index_unclassified]<-"Unclassified due to insufficient info"

```


```{r Dataframe where all the sample subtype are determined based on TPM cutoff}
index_clincal<- index_clincal[2:length(index_clincal)] # Remove NA at index 1

TPM_index <- c(index_clincal,index_tpm_unsubtype_above,index_tpm_unsubtype_below)

df_tpm <- alteration_df[TPM_index,]

df_tpm <- df_tpm %>% select(Kids_First_Biospecimen_ID_DNA,Kids_First_Biospecimen_ID_RNA,copy_number,
                            pathology_free_text_diagnosis,status,MYCN_TPM,subtype)
```

QC Check
```{r}

### QC Check with ERIC
TARGET<-readr::read_tsv(file.path(dataDir, "histologies.tsv"),guess_max = 100000) 
GMFK<-readr::read_tsv(file.path(dataDir, "histologies.tsv"),guess_max = 100000) 

TARGET<- TARGET %>% filter(cohort=="TARGET",cancer_group %in% c("Neuroblastoma", "Ganglioneuroblastoma","Ganglioneuroma"), experimental_strategy=="WXS") 
GMFK<- GMFK %>% filter(cohort=="GMKF",cancer_group %in% c("Neuroblastoma", "Ganglioneuroblastoma","Ganglioneuroma")) 

TARGET <- TARGET %>% select(Kids_First_Biospecimen_ID,Kids_First_Participant_ID,sample_id,cohort,experimental_strategy)
GMFK <- GMFK %>% select(Kids_First_Biospecimen_ID,Kids_First_Participant_ID,sample_id,cohort,experimental_strategy)

df_merge <- inner_join(TARGET,GMFK, by= c("Kids_First_Participant_ID","sample_id"))


colnames(df_merge)<- c("Kids_First_Biospecimen_ID_TARGET",
                        "Kids_First_Participant_ID","sample_id","cohort_TARGET",
                        "experimental_strategy_Target","Kids_First_Biospecimen_ID_GMKF","cohort_GMKF",
                       "experimental_strategy_GMKF")
df_merge$TPM_GMKF<-NA
df_merge$TPM_Target<-NA

df_merge$status_GMKF<-NA
df_merge$status_Target<-NA

df_merge$subtype_GMKF<-NA
df_merge$subtype_Target<-NA

for (iter in 1:dim(df_merge)[1]){
  index_gmkf <- which(alteration_df$Kids_First_Biospecimen_ID_DNA == df_merge$Kids_First_Biospecimen_ID_GMKF[iter])
  index_target <- which(alteration_df$Kids_First_Biospecimen_ID_DNA == df_merge$Kids_First_Biospecimen_ID_TARGET[iter])

  if(length(index_gmkf) ==0 ){
    index_gmkf <- which(alteration_df$Kids_First_Biospecimen_ID_RNA == df_merge$Kids_First_Biospecimen_ID_GMKF[iter])
  }
  df_merge$TPM_GMKF[iter]<- alteration_df$MYCN_TPM[index_gmkf]
  df_merge$status_GMKF[iter]<- alteration_df$status[index_gmkf]
  df_merge$subtype_GMKF[iter]<- alteration_df$subtype[index_gmkf]
  
  if(length(index_target) ==0 ){
    index_target <- which(alteration_df$Kids_First_Biospecimen_ID_RNA == 
                            df_merge$Kids_First_Biospecimen_ID_TARGET[iter])
  }
  df_merge$TPM_Target[iter]<- alteration_df$MYCN_TPM[index_target]
  df_merge$status_Target[iter]<- alteration_df$status[index_target]
  df_merge$subtype_Target[iter]<- alteration_df$subtype[index_target]
}
```

#Part 6: QC Check

```{r}


TARGET2<- hist_df %>% filter(cohort=="TARGET",pathology_diagnosis %in% c("Neuroblastoma", "Ganglioneuroblastoma","Ganglioneuroblastoma, nodular","Ganglioneuroblastoma, intermixed","Ganglioneuroma, maturing subtype OR Ganglioneuroblastoma, well differentiated"),
                            experimental_strategy=="WXS") 
GMKF2<- hist_df %>% filter(cohort=="GMKF",pathology_diagnosis %in% c("Neuroblastoma", "Ganglioneuroblastoma","Ganglioneuroblastoma, nodular","Ganglioneuroblastoma, intermixed","Ganglioneuroma, maturing subtype OR Ganglioneuroblastoma, well differentiated"), experimental_strategy=="WGS") 

TARGET2 <- TARGET2 %>% select(Kids_First_Biospecimen_ID,Kids_First_Participant_ID,sample_id,
                              cohort,experimental_strategy)
GMKF2 <- GMKF2 %>% select(Kids_First_Biospecimen_ID,Kids_First_Participant_ID,sample_id,
                          cohort,experimental_strategy)

QC_df <- inner_join(GMKF2,TARGET2, by= c("Kids_First_Participant_ID","sample_id"))

QC_df <-  QC_df %>% select(Kids_First_Biospecimen_ID.x,Kids_First_Biospecimen_ID.y,Kids_First_Participant_ID,sample_id,cohort.x,cohort.y,experimental_strategy.x,experimental_strategy.y)

colnames(QC_df)<- c("Kids_First_Biospecimen_ID_GMKF","Kids_First_Biospecimen_ID_TARGET",
                        "Kids_First_Participant_ID","sample_id","cohort_GMKF","cohort_TARGET",
                        "experimental_strategy_GMKF","experimental_strategy_Target")

QC_df$TPM_GMKF<-NA
QC_df$TPM_Target<-NA

QC_df$status_GMKF<-NA
QC_df$status_Target<-NA

QC_df$subtype_GMKF<-NA
QC_df$subtype_Target<-NA

for (iter in 1:dim(QC_df)[1]){
  index_gmkf <- which(alteration_df$Kids_First_Biospecimen_ID_DNA == QC_df$Kids_First_Biospecimen_ID_GMKF[iter])
  index_target <- which(alteration_df$Kids_First_Biospecimen_ID_DNA == QC_df$Kids_First_Biospecimen_ID_TARGET[iter])

  if(length(index_gmkf) ==0 ){
    index_gmkf <- which(alteration_df$Kids_First_Biospecimen_ID_RNA == QC_df$Kids_First_Biospecimen_ID_GMKF[iter])
  }
  QC_df$TPM_GMKF[iter]<- alteration_df$MYCN_TPM[index_gmkf]
  QC_df$status_GMKF[iter]<- alteration_df$status[index_gmkf]
  QC_df$subtype_GMKF[iter]<- alteration_df$subtype[index_gmkf]
  
  if(length(index_target) ==0 ){
    index_target <- which(alteration_df$Kids_First_Biospecimen_ID_RNA == 
                            QC_df$Kids_First_Biospecimen_ID_TARGET[iter])
  }
  QC_df$TPM_Target[iter]<- alteration_df$MYCN_TPM[index_target]
  QC_df$status_Target[iter]<- alteration_df$status[index_target]
  QC_df$subtype_Target[iter]<- alteration_df$subtype[index_target]
}

```



#Part 7: Write the table
```{r write alteration table}
alteration_df%>%readr::write_tsv(file.path(results_dir, "Alteration_Table.tsv"))
df_tpm%>%readr::write_tsv(file.path(results_dir, "Subtypes_Based_On_Cutoff.tsv"))
QC_df%>%readr::write_tsv(file.path(results_dir, "QC_table.tsv"))
df_merge%>%readr::write_tsv(file.path(results_dir, "QC_table2.tsv"))
```



# References
[1]https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/independent-samples/util/independent-rna-samples.R

[2]https://www.codingprof.com/3-easy-ways-to-remove-empty-columns-in-r/#:~:text=Identify%20the%20empty%20columns,empty%20values%20in%20a%20column.

[3] https://stat.ethz.ch/pipermail/r-help/2010-April/234706.html

[4]https://stackoverflow.com/questions/5234117/how-to-drop-columns-by-name-in-a-data-frame

[5]https://stackoverflow.com/questions/64638958/grouped-bars-in-descending-order