---
title: "Identify samples for patients with both methylation and RNA-Seq data"
output: html_notebook
author: Eric Wafula for Pedaitric Open Target
date: 2023
---

To run and fully test the upstream `post array preprocessing` modules in continuous integration, we must ensure that there are examples of samples for patients with both methylation and RNA-Seq data in both methylation beta/m-value and RNA-Seq gene/isoform expression subset matrices respectivey.

```{r}
suppressPackageStartupMessages(library(tidyverse))
options(readr.show_col_types = FALSE)
```

```{r}
set.seed(2022)
```

The histologies file contains the information we need for identifying samples for patients with both methylation and RNA-Seq data in a cancer_group-cohort combination.

```{r}
histologies_df <- read_tsv(file.path("..", "..", "data", "histologies.tsv")) %>% 
 filter(sample_type == "Tumor" & 
          tumor_descriptor == "Primary Tumor" & 
          (experimental_strategy == "RNA-Seq" | 
             experimental_strategy == "Methylation")) %>% 
  select(Kids_First_Biospecimen_ID, Kids_Participant_ID, sample_id, cohort) %>% 
  mutate(match_id = paste(Kids_Participant_ID, sample_id, sep = "_" )) %>% 
  select(-Kids_Participant_ID, -sample_id, sample_id)
  
```
### Methylation samples

Get methyl biospecimen identifiers in methyl independent sample list

```{r}
independent_samples <- read_tsv(
  file.path("..", "..", "data", 
            "independent-specimens.methyl.primary.eachcohort.tsv"))

methyl_samples <- histologies_df %>% 
  filter(experimental_strategy == "Methylation", 
         Kids_First_Biospecimen_ID  %in% 
           independent_samples$Kids_First_Biospecimen_ID) %>% 
  rename(Kids_First_Biospecimen_ID_Methly = Kids_First_Biospecimen_ID)

```

### RNA-Seq samples 

Get rnaseq biospecimen identifiers in rnaseq independent sample list

```{r}
independent_samples <- read_tsv(
  file.path("..", "..", "data", 
            "independent-specimens.methyl.primary.eachcohort.tsv"))

rnaseq_samples <- histologies_df %>% 
  filter(experimental_strategy == "RNA-Seq", 
         Kids_First_Biospecimen_ID  %in% 
           independent_samples$Kids_First_Biospecimen_ID) %>% 
   rename(Kids_First_Biospecimen_ID_RNA = Kids_First_Biospecimen_ID)
```

### TARGET _Methylation_ examples

TARGET biospecimen identifiers for patients with both methylation and RNA-Seq data

```{r}
methyl_rnaseq_samples <- inner_join(methyl_samples, by = c(match_id, cohort)) %>% 
  filter(cohort == "TARGET") %>% 
  
```

```{r}
# randomly select TARGET poly-A MYCN amplified samples 
mycn_examples %>%
  filter(Kids_First_Biospecimen_ID %in% 
           sample(mycn_examples %>% 
                    filter(molecular_subtype == "MYCN amp", 
                           RNA_library == "poly-A") %>% 
                    pull(Kids_First_Biospecimen_ID) %>% 
                    unique(), 
                  size = 5)) %>% 
  arrange(Kids_First_Biospecimen_ID)
```


### TARGET _poly-A MYCN non-amp_ RNA-Seq examples

```{r}
# randomly select TARGET poly-A MYCN non-amplified samples
mycn_examples %>%
  filter(Kids_First_Biospecimen_ID %in% 
           sample(mycn_examples %>% 
                    filter(molecular_subtype == "MYCN non-amp",
                           RNA_library == "poly-A") %>% 
                    pull(Kids_First_Biospecimen_ID) %>% 
                    unique(), 
                  size = 5)) %>% 
  arrange(Kids_First_Biospecimen_ID)
```

### PBTA _stranded Diffuse midline glioma_ RNA-Seq examples

```{r}
# randomly select PBTA stranded Diffuse midline glioma samples
glioma_examples %>%
  filter(Kids_First_Biospecimen_ID %in% 
           sample(glioma_examples %>% 
                    filter(cancer_group == "Diffuse midline glioma",
                           RNA_library == "stranded") %>% 
                    pull(Kids_First_Biospecimen_ID) %>% 
                    unique(), 
                  size = 5)) %>% 
  arrange(Kids_First_Biospecimen_ID)
```

### PBTA _poly-A Diffuse midline glioma_ RNA-Seq examples

```{r}
# randomly select PBTA poly-A Diffuse midline glioma samples
glioma_examples %>%
  filter(Kids_First_Biospecimen_ID %in% 
           sample(glioma_examples %>% 
                    filter(cancer_group == "Diffuse midline glioma",
                           RNA_library == "poly-A") %>% 
                    pull(Kids_First_Biospecimen_ID) %>% 
                    unique(), 
                  size = 5)) %>% 
  arrange(Kids_First_Biospecimen_ID)
```

### PBTA _stranded High-grade glioma/astrocytoma_ RNA-Seq examples

```{r}
# randomly select PBTA stranded High-grade glioma/astrocytoma samples
glioma_examples %>%
  filter(Kids_First_Biospecimen_ID %in% 
           sample(glioma_examples %>% 
                    filter(cancer_group == "High-grade glioma/astrocytoma",
                           RNA_library == "stranded") %>% 
                    pull(Kids_First_Biospecimen_ID) %>% 
                    unique(), 
                  size = 5)) %>% 
  arrange(Kids_First_Biospecimen_ID)
```

### PBTA _poly-A High-grade glioma/astrocytoma_ RNA-Seq examples

```{r}
# randomly select PBTA poly-A High-grade glioma/astrocytoma samples
glioma_examples %>%
  filter(Kids_First_Biospecimen_ID %in% 
           sample(glioma_examples %>% 
                    filter(cancer_group == "High-grade glioma/astrocytoma",
                           RNA_library == "poly-A") %>%
                    pull(Kids_First_Biospecimen_ID) %>% 
                    unique(), 
                  size = 5)) %>% 
  arrange(Kids_First_Biospecimen_ID)
```

### GTEx _Brain Cortex_ RNA-Seq examples

```{r}
# randomly select GTEx Brain Cortex samples
gtex_examples %>%
  filter(Kids_First_Biospecimen_ID %in% 
           sample(gtex_examples %>% 
                    filter(gtex_subgroup == "Brain - Cortex") %>%
                    pull(Kids_First_Biospecimen_ID) %>% 
                    unique(), 
                  size = 10)) %>% 
  arrange(Kids_First_Biospecimen_ID)
```

### GTEx _Brain Cerebellum_ RNA-Seq examples

```{r}
# randomly select GTEx Brain Cerebellum samples
gtex_examples %>%
  filter(Kids_First_Biospecimen_ID %in% 
           sample(gtex_examples %>% 
                    filter(gtex_subgroup == "Brain - Cerebellum") %>%
                    pull(Kids_First_Biospecimen_ID) %>% 
                    unique(), 
                  size = 10)) %>% 
  arrange(Kids_First_Biospecimen_ID)
```

### Session Info

```{r}
sessionInfo()
```

