---
title: "Examine `tumor_descriptor` and `experimental_strategy` distributions"
author: "ZZGeng"
date: "2024-06-03"
output:
  html_document:
    df_print: paged
  pdf_document: default
---


This notebook is adapted from OpenPBTA (https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/sample-distribution-analysis/03-tumor-descriptor-and-assay-count.Rmd) to investigate the distribution of experimental strategy of OpenPedCan project. 

```{r}
library(tidyverse)
# this library will help display tables with smaller font for easy viewing
# in the HTML setting
library(flextable)
```

## Read in histologies file

`pbta-histologies.tsv` contains all the clinical information.

```{r}
root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
analysis_dir <- file.path(root_dir, "figures", "manuscript_OPC")
results_dir <- file.path(analysis_dir, "results")
plots_dir <- file.path(analysis_dir, "plots")

```

## read histologies
```{r}
histology_file <- file.path(root_dir, "data", "histologies.tsv")
histology_df <- readr::read_tsv(histology_file, guess_max = 10000)
```

For WGS and WXS samples, we'll have tumor and normal to get the somatic calls.
We'll limit this to tumor samples to avoid double-counting participants and we'll remove derived cell lines.

```{r}
tumor_df <- histology_df %>%
  filter(sample_type == "Tumor",
         composition == "Solid Tissue") 
```

## Number of each assay

First, we'll examine how many of each type of assay we have (tumors only).
This information is stored in the `experimental_strategy` column.

```{r}
exp_strategy_df <- tumor_df %>%
  group_by(experimental_strategy) %>%
  tally() %>%
  arrange(desc(n)) 

exp_strategy_df %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")
```

## generate plot for experimental strategy
```{r}
exp_strategy_df %>%
  ggplot2::ggplot(ggplot2::aes(x = factor(experimental_strategy, levels = (experimental_strategy)), 
                               y = n, fill = n)) +
  ggplot2::geom_col() +
  ggplot2::theme_bw() +
  ggplot2::labs(x = "Experimental Strategy", y = "Count",
                title = "Sample Distribution Across Experimental Strategy") +
  ggplot2::theme(axis.text.x = ggplot2::element_text(
    angle = 75,
    hjust = 1,
    size = 8
  ),
  panel.grid = ggplot2::element_blank()) 
```

```{r}
ggsave(
  file = file.path(plots_dir, "distribution_across_experimental_strategy.pdf"),
  width = 12,
  height = 10)
```

## Primary vs. other (recurrence, progressive disease)

We'll look at the breakdown of the `tumor_descriptor` column, separating the genomic assays from transcriptomic assays.

```{r}
tumor_descriptor_df <- tumor_df %>%
  select(Kids_First_Participant_ID, 
         experimental_strategy, 
         tumor_descriptor) %>%
  arrange(Kids_First_Participant_ID) 
```

### Genomic assays

Setting aside the `Panel` sample for the moment and only looking at WXS and WGS assays.
We're collapsing the different values in `tumor_descriptor` to form a single descriptor when there are multiple types of tumors from the same individual.

```{r}
genomic_df <- tumor_descriptor_df %>%
  filter(experimental_strategy %in% c("WGS", "WXS")) %>%
  group_by(Kids_First_Participant_ID) %>%
  summarize(descriptors = paste(sort(unique(tumor_descriptor)),  
                                collapse = ", "),
            experimental_strategy = paste(sort(unique(experimental_strategy)),
                                          collapse = ", "))
```

```{r}
genomic_df %>%
  group_by(descriptors) %>%
  tally() %>%
  arrange(desc(n)) %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")
```

Only primary samples is the most common case for genomic assays.

### Transcriptomic assays

Looking only at RNA-seq samples and performing the same collapsing of the `tumor_descriptor` column.

```{r}
transcriptomic_df <- tumor_descriptor_df %>%
  filter(experimental_strategy == "RNA-Seq") %>%
  group_by(Kids_First_Participant_ID) %>%
  summarize(descriptors = paste(sort(unique(tumor_descriptor)),  
                                collapse = ", "),
            experimental_strategy = unique(experimental_strategy))
```
 
 
 
```{r}
transcriptomic_df %>%
  group_by(descriptors) %>%
  tally() %>%
  arrange(desc(n)) %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")
```

The same holds for RNA-seq.

### Paired genomic, transcriptomic assays

How many participants have paired genomic and transcriptomic samples for the same `tumor_descriptor` values?

```{r}
# if we perform a full join here using the pariticpant ID and descriptors, we 
# will get NAs in columns where pairs don't exist and can use this information 
# to count
paired_df <-
  full_join(genomic_df, transcriptomic_df,
                   by = c("Kids_First_Participant_ID", "descriptors"))
```

#### All paired

Count the examples where all time points have both kinds of assays.

```{r}
# no NAs = both kinds of assays a re present
paired_df %>%
  filter(complete.cases(.)) %>%
  group_by(descriptors) %>%
  tally() %>%
  arrange(desc(n)) %>% 
  regulartable() %>%
  fontsize(size = 12, part="all")
```

#### No RNA-seq

If the `experimental_strategy.y` column has an `NA`, RNA-seq is missing for that participant-descriptors pair.

```{r}
paired_df %>%
  filter(is.na(experimental_strategy.y)) %>%
  group_by(descriptors) %>%
  tally() %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")
```

#### No WXS or WGS

If the `experimental_strategy.x` column has an `NA`, there is no WGS or WXS for that participant-descriptors pair.

```{r}
paired_df %>%
  filter(is.na(experimental_strategy.x)) %>%
  group_by(descriptors) %>%
  tally() %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")
```

#### Duplicate participant IDs

There are cases where one timepoint (e.g., `tumor_descriptor` value) is missing from one kind of assay but not the other.
This will show up as duplicated values in the `Kids_First_Participant_ID` column.

```{r}
ids_with_dups <- paired_df %>%
  filter(duplicated(Kids_First_Participant_ID)) %>%
  pull(Kids_First_Participant_ID)
```

There are `r length(ids_with_dups)` cases of this.
Let's see what this looks like.

```{r}
paired_df %>%
  filter(Kids_First_Participant_ID %in% ids_with_dups) %>%
  arrange(Kids_First_Participant_ID)  %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")
```

*Two examples for interpretation:*

So for the `PT_3KM9W8S8` participant ID, there is primary WGS and RNA-seq data, but only RNA-seq data for progressive disease.
For `PT_HT4HJXY6`, there is WGS data for both the primary CNS tumor and the second malignancy, but RNA-seq data only for the second malignancy.

## By histology

We're going to use the `harmonized_diagnosis` column here.

### Primary only

We're including *any* assay type.
This table is different than what is plotted upstream in this module because we didn't restrict that to `Initial CNS Tumor` only.

```{r}
tumor_df %>%
  filter(tumor_descriptor == "Initial CNS Tumor") %>%
  distinct(Kids_First_Participant_ID, harmonized_diagnosis) %>%
  group_by(harmonized_diagnosis) %>%
  tally() %>%
  arrange(desc(n)) %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")
```

### Disease type - descriptors pairs

```{r}
disease_types_descriptors <- tumor_df %>%
  group_by(Kids_First_Participant_ID) %>%
  summarize(disease_types = paste(sort(unique(harmonized_diagnosis)),
                                  collapse = ", "),
            descriptors = paste(sort(unique(tumor_descriptor)),
                                collapse = ", ")) %>%
  group_by(disease_types, descriptors) %>%
  tally() %>%
  arrange(desc(n))

disease_types_descriptors %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")
```

#### Remove primary only

```{r}
disease_types_descriptors %>%
  filter(descriptors != "Initial CNS Tumor")  %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")
```

What about when the primary tumor is present and paired with another point in time?

```{r}
disease_types_descriptors %>%
  filter(descriptors != "Initial CNS Tumor")  %>% 
  filter(grepl("Initial CNS Tumor", descriptors)) %>%
  regulartable() %>%
  fontsize(size = 12, part = "all")
```