---
title: "High-Grade Glioma Molecular Subtyping - Combine DNA Assays"
author: "Chante Bethell, Jaclyn Taroni for ALSF CCDL, Zhuangzhuang Geng for D3b, Eric Wafula for DBHI, Jo Lynne Rokita for D3b"
date: "2020"
output:
  html_notebook:
    toc: TRUE
    toc_float: TRUE
---

This notebook joins copy number alteration, mutation, fusion and gene expression data for the
purposes of subtyping HGG samples
([`AlexsLemonade/OpenPBTA-analysis#249`](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/249)).

All data are cleaned in notebooks upstream:

* Copy number data in [`03-HGG-molecular-subtyping-cnv.Rmd`](./03-HGG-molecular-subtyping-cnv.Rmd)
* Mutation data in [`04-HGG-molecular-subtyping-mutation.Rmd`](./04-HGG-molecular-subtyping-mutation.Rmd); specific defining lesions are prepared in [`01-HGG-molecular-subtyping-defining-lesions.Rmd`](./01-HGG-molecular-subtyping-defining-lesions.Rmd)
* Fusion data in [`05-HGG-molecular-subtyping-fusion.Rmd`](./05-HGG-molecular-subtyping-fusion.Rmd)
* Gene expression data in [`06-HGG-molecular-subtyping-gene-expression.Rmd`](./06-HGG-molecular-subtyping-gene-expression.Rmd)

## Usage

This notebook is intended to be run via the command line from the top directory
of the repository as follows:

```
Rscript -e "rmarkdown::render('analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-dna.Rmd', clean = TRUE)"
```

## Set up

### Libraries and Functions

```{r}
library(tidyverse)
```

### Directories

```{r}
# Detect the ".git" folder -- this will in the project root directory.
# Use this as the root directory to ensure proper sourcing of functions no
# matter where this is called from
root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))

# File path to results directory -- this contains the cleaned data
results_dir <-
  file.path(root_dir, "analyses", "molecular-subtyping-HGG", "results")

data_dir <- file.path(root_dir, "data")
```

### Read in Files

```{r message=FALSE}
cn_gistic_df <- read_tsv(file.path(results_dir, "HGG_cleaned_cnv.tsv"))
mutation_df <- read_tsv(file.path(results_dir, "HGG_cleaned_mutation.tsv"))
histologies_df <- read_tsv(file.path(data_dir, "histologies-base.tsv"), guess_max = 100000) %>% 
  dplyr::filter(cohort %in% c("PBTA", "DGD")) %>%
  # create cell line merge id
  mutate(cell_line_id = paste(cell_line_composition, cell_line_passage, sep = "_"))
fusion_df <- read_tsv(file.path(results_dir, "HGG_cleaned_fusion.tsv"))
exp_df <- read_tsv(file.path(results_dir,
                                "HGG_cleaned_expression.tsv"))

# diff in exp and fusion files?
missing_from_exp <- setdiff(fusion_df$Kids_First_Biospecimen_ID, exp_df$Kids_First_Biospecimen_ID)

```

### Output Files

```{r}
# full table
output_all <- file.path(results_dir, "HGG_cleaned_all_table.tsv")
# molecular subtype table
output_subtype <- file.path(results_dir, "HGG_molecular_subtype.tsv")
```

## Join Together

```{r}
dna_df <- mutation_df %>%
  rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>%
  left_join(cn_gistic_df) %>%
  left_join(histologies_df[, c("Kids_First_Participant_ID", "sample_id", "Kids_First_Biospecimen_ID", "composition", "cell_line_id")])
```

```{r}
rm(mutation_df, cn_gistic_df)
```

### Join RNA-seq Data (Fusion, Gene Expression)

We used a combined gene expressoin file in the previous module and we will continue using combined one here.

```{r}
# read in full file
putative_oncogenic_df <-
  read_tsv(file.path(root_dir,
    "data",
    "fusion-putative-oncogenic.tsv"))
```

### Join the Expression and Fusion Data

```{r}
rna_df <- left_join(exp_df, fusion_df)
head(rna_df, n = 10)
```

```{r}
rna_df <- histologies_df %>%
  select(Kids_First_Participant_ID,
         sample_id,
         Kids_First_Biospecimen_ID, 
         composition, cell_line_id) %>%
  inner_join(rna_df)
```

```{r}
rm(exp_df)
```

### All Data

Create one table with the all cleaned data.

```{r}

all_data_df <- full_join(dna_df, rna_df,
                         by = c("Kids_First_Participant_ID",
                                "sample_id","cell_line_id"),
                         suffix = c("_DNA", "_RNA"))

head(all_data_df, n = 10)
```

Add in brain region and age.

```{r}
relevant_clinical_df <- histologies_df %>%
  filter(sample_id %in% all_data_df$sample_id,
         sample_type == "Tumor") %>%
  mutate(age_at_diagnosis_yr =
           floor(as.integer(age_at_diagnosis_days)/365)) %>%
  group_by(Kids_First_Participant_ID, sample_id) %>%
  summarize(CNS_region = paste(sort(unique(CNS_region)),
                               collapse = ", "),
            age_at_diagnosis_yr = paste(sort(unique(age_at_diagnosis_yr)),
                                        collapse = ", "))
```

Merge DNA and RNA compotistion.

```{r}
all_data_df <- inner_join(relevant_clinical_df,
                          all_data_df) %>%
  mutate(composition = case_when(composition_DNA == composition_RNA ~ composition_DNA, 
                                 is.na(composition_DNA)~composition_RNA, 
                                 is.na(composition_RNA)~composition_DNA)) %>% 
  select(-c("composition_DNA", "composition_RNA")) %>%
  dplyr::arrange(Kids_First_Participant_ID, sample_id)
```

Add methylation to all_data_df.

```{r}
methyl_df <- histologies_df %>% 
  filter(experimental_strategy == "Methylation", 
         ((grepl("GBM_", dkfz_v12_methylation_subclass) |
             grepl("pedHGG_", dkfz_v12_methylation_subclass) | 
             dkfz_v12_methylation_subclass == "DHG_G34" | 
             dkfz_v12_methylation_subclass == "DMG_K27" | 
             dkfz_v12_methylation_subclass == "A_IDH_HG" |
             dkfz_v12_methylation_subclass == "IHG") & 
          dkfz_v12_methylation_subclass_score >= 0.8)) %>% 
  mutate(molecular_subtype_methyl = case_when(
    grepl("GBM_", dkfz_v12_methylation_subclass) ~ "HGG, H3 wildtype",
    grepl("pedHGG_", dkfz_v12_methylation_subclass) ~ "HGG, H3 wildtype",
    dkfz_v12_methylation_subclass == "DHG_G34"  ~ "HGG, H3 G35",
    dkfz_v12_methylation_subclass == "DMG_K27"  ~ "DMG, H3 K28",
    dkfz_v12_methylation_subclass == "A_IDH_HG"  ~ "HGG, IDH",
    dkfz_v12_methylation_subclass == "IHG"  ~ "IHG")) %>% 
  select(Kids_First_Biospecimen_ID, Kids_First_Participant_ID, sample_id, 
                composition, molecular_subtype_methyl) %>%
  rename("Kids_First_Biospecimen_ID_Methyl" = "Kids_First_Biospecimen_ID") %>% 
  distinct()


all_data_df <-  all_data_df %>% 
  full_join(methyl_df, by = c("Kids_First_Participant_ID", "sample_id", "composition"))  %>% 
  select(starts_with("Kids_First"), sample_id, everything())

```

Add IHG biospecimen from pathology_diagnosis_free_text to all_data_df.

```{r}
igh_path_free_dx <- histologies_df %>% 
  filter(experimental_strategy != "Methylation",
         grepl("infant type hemispheric glioma", pathology_free_text_diagnosis))

ihg_dna <- igh_path_free_dx %>% 
  filter((experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing") & is.na(RNA_library)),
         !Kids_First_Biospecimen_ID %in% all_data_df$Kids_First_Biospecimen_ID_DNA) %>% 
  select(Kids_First_Biospecimen_ID, sample_id, Kids_First_Participant_ID, composition) %>% 
  rename(Kids_First_Biospecimen_ID_DNA = Kids_First_Biospecimen_ID)

ihg_rna <- igh_path_free_dx %>% 
  filter((experimental_strategy %in% c("RNA-Seq", "Targeted Sequencing") & !is.na(RNA_library)),
         !Kids_First_Biospecimen_ID %in% all_data_df$Kids_First_Biospecimen_ID_RNA) %>% 
  select(Kids_First_Biospecimen_ID, sample_id, Kids_First_Participant_ID, composition) %>% 
  rename(Kids_First_Biospecimen_ID_RNA = Kids_First_Biospecimen_ID)

all_data_df <-  all_data_df %>% full_join(ihg_dna) %>% full_join(ihg_rna) %>% 
  mutate(clinical_pathology = 
           case_when(Kids_First_Biospecimen_ID_DNA %in% 
                       ihg_dna$Kids_First_Biospecimen_ID_DNA ~ "IHG",
                     Kids_First_Biospecimen_ID_RNA %in% 
                       ihg_rna$Kids_First_Biospecimen_ID_RNA ~ "IHG")) %>% 
  select(starts_with("Kids_First"), sample_id, everything())

```

Write to file.

```{r}
write_tsv(all_data_df, output_all)
```

## Tables by Molecular Subtype

In this section, we'll divide up the table based on the molecular subtypes described in [`AlexsLemonade/OpenPBTA-analysis#249`](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/249).

### H3 K28 mutant

> * These tumors contain H3F3A K28M or HIST1H3B K28M mutations
> * Co-occurring lesions include: ACVR1, TP53, ATRX mutations; PDGFRA amplification; PTEN loss
> * Mutually-exclusive lesions: FGFR1 mutations/fusions (thalamic); IDH1 mutations
> * Average age of 9 years
> * Majority should be midline localized

```{r}
h3_k28_df <- all_data_df %>%
  # Only rows with H3 K28M mutations
  filter("H3-3A.K28M" == "Yes" | H3C2.K28M == "Yes" |
           H3C3.K28M == "Yes" | H3C14.K28M == "Yes") %>%
  # Only the relevant columns
  select(Kids_First_Participant_ID,
         sample_id,
         "H3-3A.K28M",
         H3C2.K28M,
         H3C3.K28M,
         H3C14.K28M,
         "H3-3A.G35R",
         "H3-3A.G35V",
         age_at_diagnosis_yr,
         CNS_region,
         relevant_coding_mutations,
         PTEN_focal_status,
         PDGFRA_focal_status,
         FGFR1_fusions) %>%
  distinct()

h3_k28_df
```

Mutually exclusive with the H3 G35 mutations?

```{r}
any(h3_k28_df$`H3-3A.G35R` == "Yes")
```

```{r}
any(h3_k28_df$`H3-3A.G35V` == "Yes")
```

Are the majority midline localized?

```{r}
table(h3_k28_df$CNS_region)
```

Yes.

```{r}
h3_k28_df %>%
  group_by(relevant_coding_mutations) %>%
  count()
```

The coding mutation list looks like what we would expect based on the description, with the exception of some BRAF mutations.
There are no _IDH1_ mutations.

```{r}
h3_k28_df %>%
  group_by(PTEN_focal_status, PDGFRA_focal_status) %>%
  count()
```

The copy number information looks a bit different than what is described, but these are not lesions that must occur.

```{r}
summary(as.integer(h3_k28_df$age_at_diagnosis_yr))
```

Age is similar to what is described.

```{r}
table(h3_k28_df$FGFR1_fusions)
```

No _FGFR1_ fusions.

The histologies file that gets distributed as part of the project and contains molecular subtypes contains one row per biospecimen.
So we'll create a table that contains the biospecimen IDs and the calls.

```{r}
h3_k28_df_biospecimen <- all_data_df %>%
  filter("H3-3A.K28M" == "Yes" | H3C2.K28M == "Yes" |
           H3C3.K28M == "Yes" | H3C14.K28M == "Yes") %>%
  mutate(molecular_subtype = "DMG, H3 K28")

```

### H3 G35 mutant

> * These tumors contain H3F3A G35R/V mutations
> * Co-occurring lesions include: ATRX/DAXX, TP53, SETD2 mutations, NTRK fusions
> * Mutually exclusive lesions: IDH1 mutations
> * Average age of 20 years

```{r}
h3_g35_df <- all_data_df %>%
  # Only rows with H3 G35 mutations
  filter("H3-3A.G35R" == "Yes" | "H3-3A.G35V" == "Yes") %>%
  # Only the relevant columns
  select(Kids_First_Participant_ID,
         sample_id,
         "H3-3A.K28M",
         H3C2.K28M,
         "H3-3A.G35R",
         "H3-3A.G35V",
         age_at_diagnosis_yr,
         relevant_coding_mutations,
         NTRK_fusions) %>%
  distinct()

h3_g35_df
```

Make a biospecimen-centric table.

```{r}
h3_g35_df_biospecimen <- all_data_df %>%
  # Only rows with H3 G35 mutations
  filter("H3-3A.G35R" == "Yes" | "H3-3A.G35V" == "Yes") %>%
  mutate(molecular_subtype = "HGG, H3 G35")
```

### IDH mutant

> * These tumors contain IDH1 R132H mutations
> * Co-occurring lesions include: TP53 mutations; P73-AS1 promoter methylation and downregulation
> * High expression of FOXG1 and OLIG2
> * Mutually exclusive lesions: chr7 gain and chr10 loss

```{r}
idh_df <- all_data_df %>%
  filter(IDH1_mutation == "p.R132H")
idh_df
```

In these instances, the broad copy data look consistent with the description, all tumors have _TP53_ mutations, and the _TP73-AS1_ expression is consistent with downregulation.

Make a biospecimen-centric table.

```{r}
idh_df_biospecimen <- all_data_df %>%
  filter(IDH1_mutation == "p.R132H") %>%
  mutate(molecular_subtype = "HGG, IDH")
```


### H3.3 and IDH wildtype

> * High-grade gliomas absent of H3F3A and IDH mutations
> * Defining lesions: MYCN, PDGFRA amplification, TP53 and TERT mutations

```{r}
wildtype_df <- all_data_df %>%
  filter("H3-3B.K28M" == "No",
         H3C2.K28M == "No",
         H3C3.K28M == "No",
         H3C14.K28M == "No",
         "H3-3B.G35R" == "No",
         "H3-3B.G35V" == "No",
         !grepl("IDH1", relevant_coding_mutations)) %>%
  select(Kids_First_Participant_ID,
         sample_id,
         relevant_coding_mutations,
         TERT_variant_classification,
         MYCN_focal_status,
         PDGFRA_focal_status) %>%
  distinct()

wildtype_df %>% arrange(MYCN_focal_status, PDGFRA_focal_status)
```

We will label samples without H3 mutations as `HGG, H3 wildtype` below.

### 1p/19q co-deleted oligodendrogliomas

> * Co-deletion of chr 1p and 19q (LOH, loss of heterozygosity of both) results in translocation t(1p;19q)
> * Nearly all have a co-occurring IDH1 mutation (R132 or R172)
> * Other co-occurring lesions include: TERT promoter, CIC, and FUBP1 mutations
> * Mutually exclusive of TP53 and ATRX mutation
> * Typically occurs in adult tumors

Let's first filter by the broad arm data and see what we get.

```{r}
all_data_df %>%
  filter(`1p` == "loss" | `19q` == "loss")
```

We don't see the _IDH1_ mutations we would expect and we do see an H3 K28M mutation, so we will leave this off as well for the moment and in a subsequent notebook we'll address [`AlexsLemonade/OpenPBTA-analysis#435 (comment)`](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/435#issuecomment-576898275):

> We expect very few, if any of these (maybe 1 when we looked by CNV), so I think maybe we can approach this cohort by starting to look for 1p/19q codeletions in all samples + IDH mutations, and if we see any, we may manually want to check the CNV plots just to confirm.


## Biospecimen Table

Join all the tables that contain the biospecimen ids and `molecular_subtype` column together.

```{r}
molecular_subtype_table <- bind_rows(h3_g35_df_biospecimen,
                                     h3_k28_df_biospecimen,
                                     idh_df_biospecimen) %>% 
  distinct()

# We'll add HGG, H3 wildtype values in cases that do not yet have labels
wildtype_df <- all_data_df %>%
  filter(molecular_subtype_methyl != "IHG" | clinical_pathology != "IHG") %>% 
  full_join(molecular_subtype_table) %>% distinct() %>%
  filter(is.na(molecular_subtype)) %>% 
  mutate(molecular_subtype = case_when(
    # molecular_subtype is NA and we didn't find any canonical histone
    # variant in the DNA sample, but molecular_subtype_methyl is not NA. So the status will be molecular_subtype_methyl
    is.na(molecular_subtype) & !is.na(Kids_First_Biospecimen_ID_DNA) & !is.na(Kids_First_Biospecimen_ID_Methyl) ~ molecular_subtype_methyl,
    # molecular_subtype is NA and we didn't find any canonical histone
    # variant in the DNA sample so the status will be "H3 wildtype"
    is.na(molecular_subtype) & !is.na(Kids_First_Biospecimen_ID_DNA) & is.na(Kids_First_Biospecimen_ID_Methyl) ~ "HGG, H3 wildtype",
    # molecular_subtype is NA, molecular_subtype_methyl is not NA, and we do not have a
    # DNA sample to check for histone variant. So status will be molecular_subtype_methyl
    is.na(molecular_subtype) & is.na(Kids_First_Biospecimen_ID_DNA) & !is.na(molecular_subtype_methyl) ~ molecular_subtype_methyl,
    # molecular_subtype is NA, molecular_subtype_methyl is NA, and we do not have a DNA sample
    # to check for histone variant. So status will be "To be classified"
    is.na(molecular_subtype) & is.na(Kids_First_Biospecimen_ID_DNA) & is.na(molecular_subtype_methyl) ~ "HGG, To be classified",
    # when molecular_subtype is not NA (we know the H3 variant subtype from DNA )
    TRUE ~ molecular_subtype)) %>% 
distinct()
```

### IHG subtype

> There is a new 2021 entity within HGGs called "Infant-type hemispheric glioma" (IHG). This HGG is cerebral (hemispheric), arises in early childhood, and is characterized by RTK (receptor tyrosine kinase) alterations, typically fusions, in the NTRK family or in ROS1, ALK, or MET.

> The subtypes are:

> IHG, NTRK-altered
> IHG, ROS1-altered
> IHG, ALK-altered
> IHG, MET-altered

```{r}
NTRK <- fusion_df %>% 
  select(Kids_First_Biospecimen_ID, NTRK_fusions) %>%
  filter(NTRK_fusions != "None") %>%
  pull(Kids_First_Biospecimen_ID) 

ALK <- fusion_df %>% 
  select(Kids_First_Biospecimen_ID, ALK_fusions) %>%
  filter(ALK_fusions != "None") %>%
  pull(Kids_First_Biospecimen_ID) 

ROS1 <- fusion_df %>% 
  select(Kids_First_Biospecimen_ID, ROS1_fusions) %>%
  filter(ROS1_fusions != "None") %>%
  pull(Kids_First_Biospecimen_ID) 

MET <- fusion_df %>% 
  select(Kids_First_Biospecimen_ID, MET_fusions) %>%
  filter(MET_fusions != "None") %>%
  pull(Kids_First_Biospecimen_ID) 

RTK_list <- c(NTRK, ALK, MET, ROS1)

IHG <- all_data_df %>% 
  filter(molecular_subtype_methyl == "IHG" | clinical_pathology == "IHG") %>% 
  full_join(molecular_subtype_table) %>% 
  full_join(wildtype_df) %>% 
  distinct() %>%
  filter(is.na(molecular_subtype)) %>%
  mutate(molecular_subtype = case_when(
    Kids_First_Biospecimen_ID_RNA %in% NTRK ~ "IHG, NTRK-altered",
    Kids_First_Biospecimen_ID_RNA %in% ALK ~ "IHG, ALK-altered",
    Kids_First_Biospecimen_ID_RNA %in% ROS1 ~ "IHG, ROS1-altered",
    Kids_First_Biospecimen_ID_RNA %in% MET ~ "IHG, MET-altered",
    !Kids_First_Biospecimen_ID_RNA %in% c(NTRK, ALK, MET, ROS1) & 
      molecular_subtype_methyl == "IHG" ~ "IHG, To be classified",
    !Kids_First_Biospecimen_ID_RNA %in% c(NTRK, ALK, MET, ROS1) &
      (molecular_subtype_methyl != "IHG" | 
         clinical_pathology == "IHG") ~ "IHG, To be classified")) %>%
  filter(!is.na(molecular_subtype)) %>% 
  distinct()

molecular_subtype_table <- bind_rows(molecular_subtype_table, wildtype_df, IHG)

```

Write to file.

```{r}
write_tsv(molecular_subtype_table, output_subtype)
```

## Session Info

```{r}
sessionInfo()
```