---
title: "Filter MTP Tables"
output: 
  html_notebook:
    toc: TRUE
    toc_float: TRUE
    toc_depth: 4
author: Eric Wafula, Sangeeta Shukla for Pediatric OpenTargets
date: 01/10/2021
---

Purpose: Remove Ensembl (ESNG) gene identifier in the mutation frequency tables, including SNV, CNV and fusion, TPM summary statistics and methylation summary tables that are not in Open Targets target list, and updated to match Open Target disease list.


## Usage

To run this from the command line, use:
```
Rscript -e "rmarkdown::render('01-filter-mtp-tables-for-current-gencode.Rmd', clean = TRUE)"
```
_This assumes you are in the modules directory of the repository, OpenPedCan-analysis/analyses/filter-mutation-frequency-tables._

## Setup

Load libraries
```{r}
# R analysis packages
suppressWarnings(
  suppressPackageStartupMessages(library(rtracklayer))
)
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(data.table))
suppressPackageStartupMessages(library(jsonlite))

# Magrittr pipe
`%>%` <- dplyr::`%>%`
```


Set up directories. 
```{r}
# directories for input and output files
root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
data_dir <- file.path(root_dir, "data")
scratch_dir <- file.path(root_dir, "scratch")
analyses_dir <- file.path(root_dir, "analyses")
module_dir <- file.path(analyses_dir, "filter-mtp-tables")
mtp_annot_results_dir <- file.path(analyses_dir,  "mtp-annotations/results")
gene_match_input_dir <- file.path(analyses_dir,  "gene_match/input")
input_dir <- file.path(scratch_dir,  "mtp-commit")
output_dir <- file.path(scratch_dir,  "mtp-filtered")
results_dir <- file.path(module_dir, "results")

# Create results folder if it doesn't exist
if (!dir.exists(results_dir)) {
  dir.create(results_dir)
}
```

## Get input data

Get Open Targets GENCODE Ensembl identifiers
```{r}
# filter for gene rows and get ids
ensg_ids <- readr::read_tsv(file.path(mtp_annot_results_dir, 
                                      "mtp-targets-mapping.tsv.gz")) %>% 
  dplyr::pull(gene_id) %>% 
  unique() 
```

Get PMTL Ensembl identifiers
```{r}
# filter for gene rows and get ids
pmtl_ids <- readr::read_tsv(file.path(gene_match_input_dir, "PMTL_v3.1.tsv")) %>% 
  dplyr::filter(Ensembl_ID != "Symbol_Not_Found") %>% 
  dplyr::pull(Ensembl_ID) %>% 
  unique() 
```

Get OpenPedCan sample IDs (Kids_First_Biospecimen_ID)
```{r}
sample_ids <- readr::read_tsv(file.path(data_dir, "histologies.tsv"), 
                              guess_max = 10000) %>% 
  dplyr::filter(sample_type == "Tumor") %>% 
  dplyr::select(Kids_First_Biospecimen_ID, cohort)
```

## Filter mutation frequencies tables

Function to filter for required Ensembl gene identifiers and write to file
```{r}

filter_mtp_table <- function(mtp_table, ensg_ids, file_name) {
  if ("Gene_Ensembl_ID" %in% colnames(mtp_table)) { 
    # filter mtp table and write to file
    mtp_table <- mtp_table %>%
      dplyr::filter(Gene_Ensembl_ID %in% ensg_ids) %>%
      dplyr::mutate(EFO = 
                      case_when(Disease == "Chordoma" ~ "MONDO_0008978",
                                Disease == "Meningioma" ~ "MONDO_0016642",
                                Disease == "Perineuroma" ~ "MONDO_0019404",
                                Disease == "Wilms tumor" ~ "MONDO_0006058",
                                TRUE ~ EFO)) %>% 
      dplyr::mutate(MONDO = 
                      case_when(Disease == "Perineuroma" ~ "MONDO_0019404",
                                TRUE ~ MONDO)) %>% 
      readr::write_tsv(file.path(output_dir, file_name))
  } else {
    # filter tsv mtp table
    mtp_table <- mtp_table %>% 
      dplyr::filter(targetFromSourceId %in% ensg_ids) %>%
      dplyr::mutate(diseaseFromSourceMappedId = 
                      case_when(Disease == "Chordoma" ~ "MONDO_0008978",
                                Disease == "Meningioma" ~ "MONDO_0016642",
                                Disease == "Perineuroma" ~ "MONDO_0019404",
                                Disease == "Wilms tumor" ~ "MONDO_0006058", 
                                TRUE ~ diseaseFromSourceMappedId)) %>% 
      dplyr::mutate(MONDO = 
                      case_when(Disease == "Perineuroma" ~ "MONDO_0019404",
                                TRUE ~ MONDO)) %>% 
      readr::write_tsv(file.path(output_dir, file_name))
     if (grepl("methyl", file_name)) {
       rds_file <- paste(unlist(str_split(file_name, "\\."))[1], "rds", sep = ".")
       mtp_table %>% readr::write_rds(file.path(output_dir, rds_file), compress = "gz")
     }
  }
return(mtp_table)
}
```

#### Gene level SNV frequencies
```{r}
# read mtp table
mtp_file <- 
  file.path(input_dir, "gene-level-snv-consensus-annotated-mut-freq.tsv.gz")
gene_level_snv <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table 
file_name <- basename(mtp_file)
filter_mtp_table(gene_level_snv, ensg_ids, file_name) 

# Ensembl gene identifiers removed
filtered_ensg <- gene_level_snv %>% 
  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(Gene_symbol, targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct()

# read the SNV consensus file
consensus_ensg <- data.table::fread(
  file.path(data_dir, "snv-consensus-plus-hotspots.maf.tsv.gz"), 
  select = c("Tumor_Sample_Barcode", "Gene"), showProgress = FALSE) %>% 
  tibble::as_tibble() %>% 
  dplyr::filter(!is.na(Gene)) %>% 
  dplyr::rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% 
  dplyr::distinct()

# merge sample IDs to filtered ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(consensus_ensg, by=c("Ensembl_ID" = "Gene"))

# merge OpenPedCan cohorts to filtered ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(sample_ids, by= "Kids_First_Biospecimen_ID")

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "gene-level-snv-consensus-annotated-mut-freq_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>%  
  table()

# Display samples in removed ENSEMBL IDs that are in PMTL list
filtered_ensg %>% dplyr::filter(Ensembl_ID %in% pmtl_ids) %>% 
  select(Ensembl_ID, Kids_First_Biospecimen_ID) %>% 
  dplyr::distinct()

# Remove data frames
rm(gene_level_snv, filtered_ensg)
```

#### Variant level SNV frequencies
```{r}
# read mtp table
mtp_file <- 
  file.path(input_dir, "variant-level-snv-consensus-annotated-mut-freq.tsv.gz")
variant_level_snv <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table
file_name <- basename(mtp_file)
filter_mtp_table(variant_level_snv, ensg_ids, file_name) 

# Ensembl gene identifiers removed
filtered_ensg <- variant_level_snv %>% 
  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(Gene_symbol, targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct()

# merge sample IDs to filtered ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(consensus_ensg, by=c("Ensembl_ID" = "Gene"))

# merge OpenPedCan cohorts to filtered ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(sample_ids, by= "Kids_First_Biospecimen_ID")

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "variant-level-snv-consensus-annotated-mut-freq_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>% 
  table()

# Display samples in removed ENSEMBL IDs that are in PMTL list
filtered_ensg %>% dplyr::filter(Ensembl_ID %in% pmtl_ids) %>% 
  select(Ensembl_ID, Kids_First_Biospecimen_ID) %>% 
  dplyr::distinct()

# Remove data frames
rm(variant_level_snv, filtered_ensg, consensus_ensg)
```

#### Gene level CNV frequencies
```{r}
# read mtp table
mtp_file <- 
  file.path(input_dir, "gene-level-cnv-consensus-annotated-mut-freq.tsv.gz")
gene_level_cnv <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table 
file_name <- basename(mtp_file)
filter_mtp_table(gene_level_cnv, ensg_ids, file_name)  

# Ensembl gene identifiers removed
filtered_ensg <- gene_level_cnv %>% 
  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(Gene_symbol, targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct()

# read the CNV consensus file
consensus_ensg <- data.table::fread(
  file.path(data_dir, "consensus_wgs_plus_cnvkit_wxs.tsv.gz"), 
  select = c("biospecimen_id", "ensembl"), showProgress = FALSE) %>% 
  tibble::as_tibble() %>% 
  dplyr::filter(!is.na(ensembl)) %>% 
  dplyr::rename(Kids_First_Biospecimen_ID = biospecimen_id) %>% 
  dplyr::distinct()

# merge sample IDs to filtered OT ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(consensus_ensg, by=c("Ensembl_ID" = "ensembl"))

# merge OpenPedCan cohorts to filtered OT ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(sample_ids, by= "Kids_First_Biospecimen_ID")

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "gene-level-cnv-consensus-annotated-mut-freq_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>% 
  table()

# Display samples in removed ENSEMBL IDs that are in PMTL list
filtered_ensg %>% dplyr::filter(Ensembl_ID %in% pmtl_ids) %>% 
  select(Ensembl_ID, Kids_First_Biospecimen_ID) %>% 
  dplyr::distinct()

# Remove data frames
rm(gene_level_cnv, filtered_ensg, consensus_ensg)
```

#### Fusion frequencies
```{r}
# read mtp table
mtp_file <- file.path(input_dir, "putative-oncogene-fusion-freq.tsv.gz")
fusion <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table 
file_name <- basename(mtp_file)
filter_mtp_table(fusion, ensg_ids, file_name) 

# Ensembl gene identifiers removed
filtered_ensg <- fusion %>% 
  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(FusionName, Gene_symbol, targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct()

# split fusion name into constituent genes  symbols
fusion_names <- data.table::fread(
  file.path(data_dir, "fusion-putative-oncogenic.tsv"), 
  select = c("Sample", "FusionName"), showProgress = FALSE) %>% 
  tibble::as_tibble() %>% 
  tidyr::separate_rows(FusionName, sep = "--", convert = FALSE) %>% 
  tidyr::separate_rows(FusionName, sep = '/', convert = FALSE) %>% 
  dplyr::rename(Kids_First_Biospecimen_ID = Sample, 
                Gene_symbol = FusionName) %>% 
  dplyr::filter(!is.na(Gene_symbol)) %>% 
  dplyr::distinct() 

# merge sample IDs to filtered OT ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(fusion_names, by="Gene_symbol") %>% 
  dplyr::select(-FusionName) %>% 
  dplyr::distinct()

# merge OpenPedCan cohorts to filtered OT ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(sample_ids, by = "Kids_First_Biospecimen_ID")

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "putative-oncogene-fusion-freq_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>% 
  table()

# Display samples in removed ENSEMBL IDs that are in PMTL list
filtered_ensg %>% dplyr::filter(Ensembl_ID %in% pmtl_ids) %>% 
  select(Ensembl_ID, Kids_First_Biospecimen_ID) %>% 
  dplyr::distinct()

# Remove frequencies data frame
rm(fusion, filtered_ensg)
```

#### Fused gene frequencies
```{r}
# read mtp table
mtp_file <- file.path(input_dir, "putative-oncogene-fused-gene-freq.tsv.gz")
fused_gene <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table 
file_name <- basename(mtp_file)
filter_mtp_table(fused_gene, ensg_ids, file_name)


# Ensembl gene identifiers removed
filtered_ensg <- fused_gene %>% 
  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(Gene_symbol, targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct()

# merge sample IDs to filtered OT ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(fusion_names, by = "Gene_symbol") %>% 
  dplyr::distinct()

# merge sample IDs to filtered OT ENSG IDs
filtered_ensg <- filtered_ensg %>% 
  dplyr::left_join(sample_ids, by = "Kids_First_Biospecimen_ID")

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "putative-oncogene-fused-gene-freq_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>% 
  table()

# Display samples in removed ENSEMBL IDs that are in PMTL list
filtered_ensg %>% dplyr::filter(Ensembl_ID %in% pmtl_ids) %>% 
  select(Ensembl_ID, Kids_First_Biospecimen_ID) %>% 
  dplyr::distinct()

# Remove frequencies data frame
rm(fused_gene, filtered_ensg, fusion_names)
```

#### Gene expression TPM group-wise summary statistics
OpenPedCan collapsed gene expressions TPM matrix only contains gene symbols 
and not ENSEMBL gene identifiers. Therefore, samples associated with removed 
ENSEMBL identifiers cannot be listed. 
```{r}
# read mtp table
mtp_file <- 
  file.path(input_dir, "long_n_tpm_mean_sd_quantile_group_wise_zscore.tsv.gz")
tpm_group_wise_stats <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table 
file_name <- basename(mtp_file)
filter_mtp_table(tpm_group_wise_stats, ensg_ids, file_name)

# Ensembl gene identifiers removed
filtered_ensg <- tpm_group_wise_stats %>% 
  dplyr::filter(!Gene_Ensembl_ID %in% ensg_ids) %>% 
  dplyr::select(Gene_symbol, Gene_Ensembl_ID, cohort) %>%
  dplyr::rename(Ensembl_ID = Gene_Ensembl_ID) %>% 
  dplyr::distinct()

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "long_n_tpm_mean_sd_quantile_group_wise_zscore_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::filter(cohort != "All Cohorts") %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>%  
  table()

# Remove summary statistics data frame
rm(tpm_group_wise_stats, filtered_ensg)
```

#### Gene expression TPM gene-wise summary statistics
OpenPedCan collapsed gene expressions TPM matrix only contains gene symbols 
and not ENSEMBL gene identifiers. Therefore, samples associated with removed 
ENSEMBL identifiers cannot be listed. 
```{r}
# read mtp table
mtp_file <- 
  file.path(input_dir, "long_n_tpm_mean_sd_quantile_gene_wise_zscore.tsv.gz")
tpm_gene_wise_stats <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table 
file_name <- basename(mtp_file)
filter_mtp_table(tpm_gene_wise_stats, ensg_ids, file_name)

# Ensembl gene identifiers removed
filtered_ensg <- tpm_gene_wise_stats %>% 
  dplyr::filter(!Gene_Ensembl_ID %in% ensg_ids) %>% 
  dplyr::select(Gene_symbol, Gene_Ensembl_ID, cohort) %>%
  dplyr::rename(Ensembl_ID = Gene_Ensembl_ID) %>% 
  dplyr::distinct()

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "long_n_tpm_mean_sd_quantile_gene_wise_zscore_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::filter(cohort != "All Cohorts") %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>%  
  table()

# Remove summary statistics data frame
rm(tpm_gene_wise_stats, filtered_ensg)
```

#### Methylation gene-level summary 
OpenPedCan contains methylation and rnaseq summaries of cohort cancer groups and
can't not associate with samples. Therefore, samples associated with removed ENSEMBL 
identifiers cannot be listed. 
```{r}
# read mtp table
mtp_file <- file.path(input_dir, "gene-methyl-beta-values-summary.tsv.gz")
gene_level_methyl <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table 
file_name <- basename(mtp_file)
filter_mtp_table(gene_level_methyl, ensg_ids, file_name)

# Ensembl gene identifiers removed
filtered_ensg <- gene_level_methyl %>% 
  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(Gene_symbol, targetFromSourceId, Dataset) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId, cohort = Dataset) %>% 
  dplyr::distinct()

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "gene-methyl-beta-values-summary_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>% 
  table()

# Remove frequencies data frame
rm(gene_level_methyl, filtered_ensg)
```

#### Methylation gene-level summary 
OpenPedCan contains methylation and rnaseq summaries of cohort cancer groups and
can't not associate with samples. Therefore, samples associated with removed ENSEMBL 
identifiers cannot be listed. 
```{r}
# read mtp table
mtp_file <- file.path(input_dir, "isoform-methyl-beta-values-summary.tsv.gz")
isoform_level_methyl <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table 
file_name <- basename(mtp_file)
filter_mtp_table(isoform_level_methyl, ensg_ids, file_name)

# Ensembl gene identifiers removed
filtered_ensg <- isoform_level_methyl %>% 
  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(Gene_symbol, targetFromSourceId, Dataset) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId, cohort = Dataset) %>% 
  dplyr::distinct()

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "isoform-methyl-beta-values-summary_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>% 
  table()

# Remove frequencies data frame
rm(isoform_level_methyl, filtered_ensg)
```

#### DESeq tumor-normal results 
OpenPedCan contains methylation and rnaseq summaries of cohort cancer groups and
can't not associate with samples. Therefore, samples associated with removed ENSEMBL 
identifiers cannot be listed. 
```{r}
# read mtp table
mtp_file <- 
  file.path(input_dir, "gene-counts-rsem-expected_count-collapsed-deseq.tsv.gz")
gene_level_deseq <- 
  data.table::fread(mtp_file, sep = "\t", showProgress = FALSE) %>%
    tibble::as_tibble() 
# filter mtp table 
file_name <- basename(mtp_file)
filter_mtp_table(gene_level_deseq, ensg_ids, file_name)

# Ensembl gene identifiers removed
filtered_ensg <- gene_level_deseq %>% 
  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(Gene_symbol, targetFromSourceId, cohort) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct()

# write to file
readr::write_tsv(filtered_ensg,
  file.path(results_dir, 
            "gene-counts-rsem-expected_count-collapsed-deseq_dropped_ensg.tsv.gz"))

# Display removed ENSEMBL IDs 
filtered_ensg %>% dplyr::select(Ensembl_ID, cohort) %>% 
  dplyr::distinct() %>%
  tidyr::drop_na() %>% 
  table()

# Remove frequencies data frame
rm(gene_level_deseq, filtered_ensg)
```

## Session Info
```{r}
sessionInfo()
```