---
title: "Tables output for manuscript"
author: "Aditya Lahiri"
date: "2021-2022"
output: 
  html_notebook:
    toc: TRUE
    toc_float: TRUE
params:
  release: v12
---

Code adapted from: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/tables/output_tables.Rmd

```{r setup, include=FALSE}
library(tidyverse)
library(openxlsx)
```

## Output Tables for OpenPedCan Manuscripts

This Rmarkdown generates tables used in the manuscript, including both main text and supplementary material. 

```{r define directories and result files}
root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
working_dir <- file.path(root_dir, "tables")
input_dir <- file.path(working_dir, "input")
analyses_dir <- file.path(root_dir, "analyses")
palette_dir <- file.path(root_dir, "figures", "palettes")

# Define data_dir based on ci. The `testing` param covers if we are in CI where 1 is CI.
if (params$release == "testing") {
  data_dir <- file.path(root_dir, "data")
} else {
  data_dir <- file.path(root_dir, "data", params$release)
}
results_dir <- file.path(working_dir, "results")
if(!dir.exists(results_dir)){
  dir.create(results_dir, recursive=TRUE)
}
```


# Define input files
```{r}
histology_file <- file.path(data_dir, "histologies.tsv")
palette_file <- file.path(palette_dir, "broad_histology_cancer_group_palette.tsv")
```

# Define output files and sheet names, when appropriate
```{r}
table_1_file <- file.path(results_dir, "Table1-molecular-subtypes.tsv")
table_s1_file <- file.path(results_dir, "TableS1-histologies.xlsx")
```

# Read files
```{r}
histology_df <- read_tsv(histology_file, guess_max =100000)
palette_df <- read_tsv(palette_file)
```
## Table 1 molecular subtypes
```{r}
subtype_n <- histology_df %>%
  # remove rows to not include
  filter(!is.na(pathology_diagnosis),
         !is.na(molecular_subtype),
         !grepl("To be classified", molecular_subtype)) %>%
   mutate(tumor_id = paste(sample_id, sample_type, composition, cell_line_composition, cell_line_passage, sep = "_"))  %>%
  # Join with palette to get broad_histology_display column
  inner_join(
    select(palette_df,
           broad_histology,
           broad_histology_display)
  ) %>%
  # remove doubled samples (WGS/RNA-seq etc.)
  distinct(broad_histology, Kids_First_Participant_ID, molecular_subtype, tumor_id) %>%
  mutate(broad_histology = case_when(broad_histology == "Ependymal tumor" ~ "Ependymoma",
         TRUE ~ broad_histology)) %>%
  dplyr::rename(`Broad histology group` = broad_histology)

# Count broad histology and subtype combinations
tumor_level <- subtype_n %>%
  select(-Kids_First_Participant_ID) %>%
  count(`Broad histology group`, molecular_subtype) %>%
  dplyr::rename(Tumors = n)

# print broad histology counts for easy check for manuscript
tumor_level %>%
  group_by(`Broad histology group`) %>%
  select(-molecular_subtype) %>%
  mutate(broad_hist_sum = sum(Tumors)) %>%
  select(-Tumors) %>%
  distinct() %>%
  arrange(desc(broad_hist_sum))

# Patient level
pt_level <- subtype_n %>%
  select(-tumor_id) %>%
  distinct() %>%
  count(`Broad histology group`, molecular_subtype) %>%
  dplyr::rename(Patients = n)

# combine patient level and tumor level tables
table1 <- pt_level %>%
  full_join(tumor_level) %>%
  dplyr::rename(`Molecular Subtype` = molecular_subtype)

# Add final row to the bottom for the total
table1 <- bind_rows(
  table1,
  tribble(
    ~`Broad histology group`, ~`Molecular Subtype`, ~Patients, ~Tumors,
    "",                       "Total",             sum(table1$Patients), sum(table1$Tumors)
  )
)

# export
write_tsv(table1, table_1_file)
```



## Table S1: histologies table

```{r}
readme <- tribble(
~`Histology column`,~Definition,~`Possible values`,
"age_at_chemo_start","Patient age at chemotherapy start in days","numeric",
"age_at_diagnosis_days","Patient age at diagnosis in days","numeric",
"age_at_event_days","Patient age at sample collection event in days","numeric",
"age_at_radiation_start","Patient age at radiation start in days","numeric",
"age_last_update_days","Patient age at the last clinical event/update in days","numeric",
"aliquot_id","External aliquot identifier","alphanumeric",
"broad_histology","Broad WHO classification of cancer type",paste(unique(histology_df$broad_histology), collapse = "; "),
"cancer_group","Harmonized cancer groupings for plots",paste(unique(histology_df$cancer_group), collapse = "; "),
"cancer_predispositions","Reported cancer predisposition syndromes",paste(unique(histology_df$cancer_predispositions), collapse = "; "),
"cell_line_composition","Cell line media",paste(unique(histology_df$cell_line_composition), collapse = "; "),
"cell_line_passage","Cell line passage at collection","numeric",
"clinical_status_at_event","Patient status at the time of sample collection", paste(unique(histology_df$clinical_status_at_event), collapse = "; "),
"CNS_region","Harmonized brain region based on `primary_site`",paste(unique(histology_df$CNS_region), collapse = "; "),
"cohort","Scientific cohort",paste(unique(histology_df$cohort), collapse = "; "),
"cohort_participant_id","Scientific cohort participant ID","C#####-C######",
"composition","Sample composition",paste(unique(histology_df$composition), collapse = "; "),
"dkfz_v11_methylation_subclass","v11 DKFZ methylation-based CNS tumor subclass","text",
"dkfz_v11_methylation_subclass_score","v11 DKFZ methylation-based CNS tumor subclass score","numeric",
"dkfz_v12_methylation_subclass","v12 DKFZ methylation-based CNS tumor subclass score","text",
"dkfz_v12_methylation_subclass_score","v12 DKFZ methylation-based CNS tumor subclass","numeric",
"dkfz_v12_methylation_mgmt_status","v12 DKFZ MGMT promoter methylation status",paste(unique(histology_df$dkfz_v11_methylation_subclass), collapse = "; "),
"dkfz_v12_methylation_mgmt_estimated","v12 DKFZ MGMT promoter methylation fraction","numeric",
"ethnicity","Patient reported ethnicity",paste(unique(histology_df$ethnicity), collapse = "; "),
"experimental_strategy","Sequencing strategy",paste(unique(histology_df$experimental_strategy), collapse = "; "),
# leaving this non-programmatic because of the duplicates that would come up (eg two selections in one patient, needing data cleanup)
"extent_of_tumor_resection","Amount of tumor resected at time of surgical event","Biopsy only;Partial resection;Gross/Near total resection;Not Reported;Unavailable",
"germline_sex_estimate","Predicted sex of patient based on germline X and Y ratio calculation (described in methods)",paste(unique(histology_df$germline_sex_estimate), collapse = "; "),
"gtex_group","Tissue Type",paste(unique(histology_df$gtex_group), collapse = "; "),
"gtex_subgroup","Tissue Subtype",paste(unique(histology_df$gtex_subgroup), collapse = "; "),
"harmonized_diagnosis","`integrated_diagnosis` if exists or updated and harmonized diagnosis using pathology_free_text_diagnosis information","text",
"integrated_diagnosis","2016 WHO diagnosis integrated from pathology diagnosis and molecular subtyping","text",
"Kids_First_Biospecimen_ID","KidsFirst biospecimen identifier","BS_########",
"Kids_First_Participant_ID","KidsFirst patient identifier","PT_########",
"molecular_subtype","Molecular subtype defined by WHO 2021 guidelines","text",
"molecular_subtype_methyl","DKFZ v12 methylation class aligned to WHO 2021 subtypes","text",
"normal_fraction","Theta2 normal DNA fraction estimate","numeric",
"Notes","Free text field describing changes from `pathology_diagnosis` to `integrated_diagnosis` or manner in which molecular_subtype was determined","text",
"OS_days","Overall survival in days","numeric",
"OS_status","Overall survival status",paste(unique(histology_df$OS_status), collapse = "; "),
"pathology_diagnosis","Reported and/or harmonized patient diagnosis from pathology reports","text",
"pathology_free_text_diagnosis","Free text patient diagnosis from pathology reports","text",
"EFS_days","Event-free survival in days","numeric",
"primary_site","Bodily site(s) from which specimen was derived","text",
"race","Patient reported race",paste(unique(histology_df$race), collapse = "; "),
"reported_gender","Patient reported gender",paste(unique(histology_df$reported_gender), collapse = "; "),
"RNA_library","Type of RNA-Sequencing library preparation",paste(unique(histology_df$RNA_library), collapse = "; "),
"sample_id","External biospecimen identifier","alphanumeric",
"sample_type","Broad sample type",paste(unique(histology_df$sample_type), collapse = "; "),
"seq_center","Sequencing center",paste(unique(histology_df$seq_center), collapse = "; "),
"short_histology","Abbreviated `cancer_group` or `broad_histology` for plotting purposes",paste(unique(histology_df$short_histology), collapse = "; "),
"tumor_descriptor","Phase of therapy from which tumor was derived",paste(unique(histology_df$tumor_descriptor), collapse = "; "),
"tumor_fraction","Theta2 tumor DNA fraction estimate","numeric",
"tumor_fraction_LUMP","LUMP tumor DNA fraction estimate from methylation","numeric",
"tumor_fraction_RFpurify_ABSOLUTE","RFpurify ABSOLUTE tumor DNA fraction estimate from methylation","numeric",
"tumor_fraction_RFpurify_ESTIMATE","RFpurify ESTIMATE tumor DNA fraction estimate from methylation","numeric",
"tumor_ploidy","Control-FREEC ploidy estimate","numeric"
)

# Combine and output
list_s1_table <- list(README = readme,
                      histologies_file = histology_df)
write.xlsx(list_s1_table, 
           table_s1_file, 
           overwrite=TRUE, 
           keepNA=TRUE)
```
