---
title: "Filter Mutation Frequencies Tables"
output: 
  html_notebook:
    toc: TRUE
    toc_float: TRUE
    toc_depth: 4
author: Eric Wafula for Pediatric OpenTargets
date: 01/10/2021
---

Purpose: Remove Ensembl (ESNG) gene identifier in the mutation frequency tables, including SNV, CNV and fusion that are not in GENCODE v38 and Ensembl package 104.


## Usage

To run this from the command line, use:
```
Rscript -e "rmarkdown::render('filter-mutation-frequencies-tables.Rmd', clean = TRUE)"
```
_This assumes you are in the modules directory of the repository, OpenPedCan-analysis/analyses/filter-mutation-frequency-tables._

## Setup

Load libraries
```{r}
# R analysis packages
suppressPackageStartupMessages(library(data.table))
suppressPackageStartupMessages(library(tidyverse))

# Magrittr pipe
`%>%` <- dplyr::`%>%`
```


Set up directories. 
```{r}
# directories for input and output files
root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
analyses_dir <- file.path(root_dir, "analyses")
module_dir <- file.path(analyses_dir, "filter-mutation-frequencies-tables")
snv_results_dir <- file.path(analyses_dir,  "snv-frequencies/results")
cnv_results_dir <- file.path(analyses_dir,  "cnv-frequencies/results")
fusion_results_dir <- file.path(analyses_dir,  "fusion-frequencies/results")
results_dir <- file.path(module_dir, "results")

# Create results folder if it doesn't exist
if (!dir.exists(results_dir)) {
  dir.create(results_dir)
}
```

## Get Ensembl gene identifiers

Load Gencode gtf file
```{r}
# gtf column names
col_names <- c("seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute")

# read file
gene_code_annot <- 
  data.table::fread(file.path(module_dir, "input", "gencode.v38.annotation.gtf.gz"),
                    col.names = col_names, skip=5, 
                    showProgress = FALSE)
```

Get Ensembl identifiers
```{r}
# filter for gene rows and get ids
ensg_ids <- gene_code_annot %>% 
  dplyr::filter(feature == "gene") %>% 
  dplyr::mutate(gene_id = stringr::str_extract(attribute, "ENSG\\d+")) %>% 
  pull(gene_id) %>% 
  unique()

# Remove gencode dataframe
rm(gene_code_annot)
```
 
## Filter mutation frequencies tables

Function to filter for required Ensembl gene identifiers and write to file
```{r}

filter_freq_table <- function(freq_file, ensg_ids) {
  
  # Read gene level table 
  freq_table <-  data.table::fread(freq_file, showProgress = FALSE) 
  
  # Write filtered TSV file
  file_name <- basename(freq_file)
  freq_table %>% dplyr::filter(targetFromSourceId %in% ensg_ids) %>% 
  data.table::fwrite(file.path(results_dir, file_name), sep = "\t", compress = "auto")

  # Write filtered JSON file
  json_file <- paste(unlist(str_split(file_name, "\\."))[1], "json", sep = ".")
  freq_table %>% dplyr::filter(targetFromSourceId %in% ensg_ids) %>% 
    jsonlite::write_json(file.path(results_dir,  json_file))  

return(freq_table)
}
```

#### Gene level SNV frequencies
```{r}
# Remove Ensembl gene identifiers
gene_level_snv <- filter_freq_table(
  file.path(snv_results_dir, "gene-level-snv-consensus-annotated-mut-freq.tsv.gz"), ensg_ids) 

# Ensembl gene identifiers removed
gene_level_snv %>% dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct() %>% 
  tibble()

# Remove frequencies dataframe
rm(gene_level_snv)
```
#### Variant level SNV frequencies
```{r}
# Remove Ensembl gene identifiers
variant_level_snv <- filter_freq_table(
  file.path(snv_results_dir, "variant-level-snv-consensus-annotated-mut-freq.tsv.gz"), ensg_ids) 

# Ensembl gene identifiers removed
variant_level_snv %>% dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct() %>% 
  tibble()

# Remove frequencies dataframe
rm(variant_level_snv)
```

#### Gene level CNV frequencies
```{r}
# Remove Ensembl gene identifiers
gene_level_cnv <- filter_freq_table(
  file.path(cnv_results_dir, "gene-level-cnv-consensus-annotated-mut-freq.tsv.gz"), ensg_ids) 

# Ensembl gene identifiers removed
gene_level_cnv %>% dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct() %>% 
  tibble()

# Remove frequencies dataframe
rm(gene_level_cnv)
```

#### Fused gene frequencies
```{r}
# Remove Ensembl gene identifiers
fused_gene <- filter_freq_table(
  file.path(fusion_results_dir, "putative-oncogene-fused-gene-freq.tsv.gz"), ensg_ids) 

# Ensembl gene identifiers removed
fused_gene %>% dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct() %>% 
  tibble()

# Remove frequencies dataframe
rm(fused_gene)
```

#### Fusion frequencies
```{r}
# Remove Ensembl gene identifiers
fusion <- filter_freq_table(
  file.path(fusion_results_dir, "putative-oncogene-fusion-freq.tsv.gz"), ensg_ids) 

# Ensembl gene identifiers removed
fusion %>% dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
  dplyr::select(targetFromSourceId) %>%
  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
  dplyr::distinct() %>% 
  tibble()

# Remove frequencies dataframe
rm(fusion)
```

## Session Info

```{r}
sessionInfo()
```